{
 "cells": [
  {
   "cell_type": "markdown",
   "source": [
    "# Get Protein Sequences\n",
    "\n",
    "Get the entire amino acid sequence for all genes targeted in endogenous datasets."
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%% md\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import requests\n",
    "import sys\n",
    "from datasets import dataset_list\n",
    "from tqdm import tqdm\n",
    "import pandas as pd\n",
    "import json\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "outputs": [],
   "source": [
    "def post_sequence_ids(ensembl_ids, data_args=None):\n",
    "    \"\"\"Take list of ensemble transcript IDs and return sequence\n",
    "\n",
    "    :param ensembl_ids: list of str\n",
    "    :param data_args: dict\n",
    "    :return: DataFrame\n",
    "    \"\"\"\n",
    "    if data_args is None:\n",
    "        data_args = {\"type\": \"protein\"}\n",
    "    data_args[\"ids\"] = ensembl_ids\n",
    "    headers= {\"content-type\" : \"application/json\", \"Accept\" : \"application/json\"}\n",
    "    server = \"https://rest.ensembl.org\"\n",
    "    ext = \"/sequence/id/\"\n",
    "    r = requests.post(server+ext, headers=headers,\n",
    "                      data=json.dumps(data_args))\n",
    "    if not r.ok:\n",
    "        r.raise_for_status()\n",
    "        sys.exit()\n",
    "    decoded = r.json()\n",
    "    sequence_df = pd.DataFrame(decoded)\n",
    "    return sequence_df\n",
    "\n",
    "\n",
    "def get_translation_overlap(ensembl_id, data_args=None):\n",
    "    if data_args is None:\n",
    "        data_args = {}\n",
    "    headers = {'content-type': 'application/json'}\n",
    "    server = \"https://rest.ensembl.org/\"\n",
    "    ext = 'overlap/translation/' + ensembl_id\n",
    "    r = requests.get(server + ext, headers=headers,\n",
    "                     data=data_args)\n",
    "    if not r.ok:\n",
    "        r.raise_for_status()\n",
    "        sys.exit()\n",
    "    decoded = r.json()\n",
    "    feature_df = pd.DataFrame(decoded)\n",
    "    return feature_df"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[{'polyphen': 0.156, 'type': 'missense_variant', 'residues': 'H/R', 'start': 806, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'allele': 'T/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'cAc/cGc', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1208461159', 'end': 806, 'Parent': 'ENST00000288602', 'sift': 0.02}, {'codons': 'Gat/Aat', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 717, 'polyphen': 0.341, 'type': 'missense_variant', 'residues': 'D/N', 'sift': 0.02, 'Parent': 'ENST00000288602', 'end': 717, 'seq_region_name': 'ENSP00000288602', 'id': 'rs886041260'}, {'sift': 0, 'end': 588, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1562954580', 'Parent': 'ENST00000288602', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 588, 'codons': 'tTt/tCt', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/G', 'polyphen': 1, 'residues': 'F/S', 'type': 'missense_variant'}, {'start': 285, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'allele': 'T/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'cAa/cGa', 'polyphen': 0.005, 'residues': 'Q/R', 'type': 'missense_variant', 'sift': 0.43, 'seq_region_name': 'ENSP00000288602', 'id': 'rs750310423', 'end': 285, 'Parent': 'ENST00000288602'}, {'residues': 'M/V', 'polyphen': 0.012, 'type': 'missense_variant', 'codons': 'Atg/Gtg', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 604, 'Parent': 'ENST00000288602', 'end': 604, 'id': 'rs745783052', 'seq_region_name': 'ENSP00000288602', 'sift': 0.01}, {'id': 'rs1257895069', 'seq_region_name': 'ENSP00000288602', 'end': 325, 'Parent': 'ENST00000288602', 'sift': 0.22, 'residues': 'S/P', 'polyphen': 0, 'type': 'missense_variant', 'start': 325, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/G', 'codons': 'Tct/Cct'}, {'start': 262, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C/G', 'codons': 'cAa/cGa/cCa', 'residues': 'Q/R/P', 'polyphen': 0.761, 'type': 'missense_variant', 'sift': 0.15, 'id': 'rs397516904', 'seq_region_name': 'ENSP00000288602', 'end': 262, 'Parent': 'ENST00000288602'}, {'Parent': 'ENST00000288602', 'id': 'rs121913361', 'seq_region_name': 'ENSP00000288602', 'end': 636, 'sift': 0, 'polyphen': 0.221, 'residues': 'G/C/R/S', 'type': 'missense_variant', 'minor_allele_frequency': None, 'allele': 'C/A/G/T', 'translation': 'ENSP00000288602', 'codons': 'Ggt/Tgt/Cgt/Agt', 'start': 636, 'clinical_significance': ['uncertain significance', 'likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation'}, {'polyphen': 0.418, 'type': 'missense_variant', 'residues': 'S/N', 'codons': 'aGt/aAt', 'minor_allele_frequency': None, 'allele': 'C/T', 'translation': 'ENSP00000288602', 'clinical_significance': ['likely pathogenic'], 'feature_type': 'transcript_variation', 'start': 645, 'Parent': 'ENST00000288602', 'end': 645, 'id': 'rs121913373', 'seq_region_name': 'ENSP00000288602', 'sift': 0.02}, {'type': 'missense_variant', 'polyphen': 0.999, 'residues': 'A/E', 'start': 2, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'allele': 'G/T', 'translation': 'ENSP00000288602', 'codons': 'gCg/gAg', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1471655119', 'end': 2, 'Parent': 'ENST00000288602', 'sift': 0}, {'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'codons': 'Gtg/Atg', 'start': 453, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'type': 'missense_variant', 'polyphen': 0.938, 'residues': 'V/M', 'sift': 0.05, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs377093637', 'end': 453}, {'sift': 0.13, 'seq_region_name': 'ENSP00000288602', 'id': 'rs543953468', 'end': 22, 'Parent': 'ENST00000288602', 'start': 22, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'minor_allele_frequency': 0.0002, 'allele': 'G/T', 'translation': 'ENSP00000288602', 'codons': 'gaC/gaA', 'polyphen': 0.995, 'residues': 'D/E', 'type': 'missense_variant'}, {'polyphen': 0.71, 'residues': 'Q/R/P', 'type': 'missense_variant', 'codons': 'cAg/cGg/cCg', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C/G', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 457, 'Parent': 'ENST00000288602', 'end': 457, 'id': 'rs978365655', 'seq_region_name': 'ENSP00000288602', 'sift': 0.07}, {'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 496, 'codons': 'caG/caT/caA', 'allele': 'C/A/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'polyphen': 0.975, 'residues': 'Q/H/Q', 'type': 'missense_variant', 'sift': 0, 'end': 496, 'id': 'rs767473452', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602'}, {'Parent': 'ENST00000288602', 'id': 'rs1425139763', 'seq_region_name': 'ENSP00000288602', 'end': 663, 'sift': 0.01, 'residues': 'E/K', 'polyphen': 1, 'type': 'missense_variant', 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Gaa/Aaa', 'start': 663, 'feature_type': 'transcript_variation', 'clinical_significance': []}, {'polyphen': 0.997, 'type': 'missense_variant', 'residues': 'G/D', 'start': 437, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'codons': 'gGt/gAt', 'id': 'rs1554400237', 'seq_region_name': 'ENSP00000288602', 'end': 437, 'Parent': 'ENST00000288602', 'sift': 0.01}, {'Parent': 'ENST00000288602', 'id': 'rs180177035', 'seq_region_name': 'ENSP00000288602', 'end': 257, 'sift': 0.02, 'residues': 'Q/R', 'polyphen': 0.46, 'type': 'missense_variant', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'codons': 'cAg/cGg', 'start': 257, 'feature_type': 'transcript_variation', 'clinical_significance': ['pathogenic']}, {'sift': 0.5, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1009904455', 'end': 408, 'Parent': 'ENST00000288602', 'start': 408, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'codons': 'cGg/cAg', 'polyphen': 0.089, 'type': 'missense_variant', 'residues': 'R/Q'}, {'residues': 'F/S', 'polyphen': 0.735, 'type': 'missense_variant', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'A/G', 'codons': 'tTt/tCt', 'start': 357, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs993830683', 'end': 357, 'sift': 0.24}, {'type': 'missense_variant', 'polyphen': 0.48, 'residues': 'R/G', 'start': 741, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'allele': 'T/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Aga/Gga', 'seq_region_name': 'ENSP00000288602', 'id': 'rs886041257', 'end': 741, 'Parent': 'ENST00000288602', 'sift': 0}, {'codons': 'cGa/cAa', 'minor_allele_frequency': None, 'allele': 'C/T', 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 598, 'polyphen': 0.999, 'residues': 'R/Q', 'type': 'missense_variant', 'sift': 0, 'Parent': 'ENST00000288602', 'end': 598, 'id': 'rs1278812236', 'seq_region_name': 'ENSP00000288602'}, {'sift': 0, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs397507485', 'end': 711, 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'codons': 'cGa/cAa', 'start': 711, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'type': 'missense_variant', 'polyphen': 0.801, 'residues': 'R/Q'}, {'type': 'missense_variant', 'polyphen': 1, 'residues': 'C/Y', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'codons': 'tGt/tAt', 'start': 572, 'clinical_significance': ['likely pathogenic'], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs397507479', 'end': 572, 'sift': 0}, {'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/G/T', 'codons': 'aGt/aCt/aAt', 'start': 784, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'polyphen': 0.769, 'type': 'missense_variant', 'residues': 'S/T/N', 'sift': 0.05, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs767671899', 'end': 784}, {'residues': 'T/A', 'polyphen': 0.001, 'type': 'missense_variant', 'codons': 'Aca/Gca', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 436, 'Parent': 'ENST00000288602', 'end': 436, 'id': 'rs1467230975', 'seq_region_name': 'ENSP00000288602', 'sift': 0.46}, {'sift': 0, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs180177041', 'end': 574, 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/G', 'codons': 'Ggc/Cgc', 'start': 574, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'polyphen': 1, 'type': 'missense_variant', 'residues': 'G/R'}, {'end': 749, 'seq_region_name': 'ENSP00000288602', 'id': 'rs397507486', 'Parent': 'ENST00000288602', 'sift': 0.01, 'polyphen': 0.992, 'type': 'missense_variant', 'residues': 'Q/R', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 749, 'codons': 'cAa/cGa', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C'}, {'type': 'missense_variant', 'polyphen': 0.062, 'residues': 'V/L', 'codons': 'Gtc/Ctc', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/G', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 805, 'Parent': 'ENST00000288602', 'end': 805, 'id': 'rs1217309134', 'seq_region_name': 'ENSP00000288602', 'sift': 0.33}, {'sift': 0.08, 'end': 442, 'id': 'rs1316280257', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 442, 'codons': 'Ccc/Acc', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/T', 'type': 'missense_variant', 'polyphen': 1, 'residues': 'P/T'}, {'sift': 0.73, 'end': 41, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1389368234', 'Parent': 'ENST00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 41, 'codons': 'Cct/Tct/Act', 'minor_allele_frequency': None, 'allele': 'G/A/T', 'translation': 'ENSP00000288602', 'residues': 'P/S/T', 'polyphen': 0.012, 'type': 'missense_variant'}, {'residues': 'N/H', 'polyphen': 0.015, 'type': 'missense_variant', 'allele': 'T/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Aac/Cac', 'start': 698, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1243095331', 'end': 698, 'sift': 0.79}, {'sift': 0.43, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1291775716', 'end': 756, 'Parent': 'ENST00000288602', 'start': 756, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'allele': 'G/C', 'translation': 'ENSP00000288602', 'codons': 'Ctg/Gtg', 'polyphen': 0.052, 'residues': 'L/V', 'type': 'missense_variant'}, {'polyphen': 0.899, 'residues': 'I/S', 'type': 'missense_variant', 'start': 503, 'clinical_significance': ['pathogenic'], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/C', 'codons': 'aTt/aGt', 'id': 'rs180177033', 'seq_region_name': 'ENSP00000288602', 'end': 503, 'Parent': 'ENST00000288602', 'sift': 0}, {'residues': 'S/G', 'polyphen': 0.003, 'type': 'missense_variant', 'codons': 'Agc/Ggc', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C', 'clinical_significance': ['likely benign'], 'feature_type': 'transcript_variation', 'start': 147, 'Parent': 'ENST00000288602', 'end': 147, 'id': 'rs397507464', 'seq_region_name': 'ENSP00000288602', 'sift': 0.35}, {'sift': 0.02, 'end': 198, 'id': 'rs756820137', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 198, 'codons': 'Tac/Cac', 'minor_allele_frequency': None, 'allele': 'A/G', 'translation': 'ENSP00000288602', 'residues': 'Y/H', 'polyphen': 0.085, 'type': 'missense_variant'}, {'end': 506, 'id': 'rs121913353', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0, 'residues': 'G/R/R', 'polyphen': 0.966, 'type': 'missense_variant', 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic'], 'start': 506, 'codons': 'Gga/Cga/Aga', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/G/T'}, {'Parent': 'ENST00000288602', 'id': 'rs758506511', 'seq_region_name': 'ENSP00000288602', 'end': 155, 'sift': 0.09, 'type': 'missense_variant', 'polyphen': 0.962, 'residues': 'P/L', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/A', 'codons': 'cCt/cTt', 'start': 155, 'clinical_significance': [], 'feature_type': 'transcript_variation'}, {'type': 'missense_variant', 'polyphen': 0, 'residues': 'A/T', 'codons': 'Gcc/Acc', 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 311, 'Parent': 'ENST00000288602', 'end': 311, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1452771445', 'sift': 0.89}, {'polyphen': 0.999, 'type': 'missense_variant', 'residues': 'R/Q', 'start': 360, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'allele': 'C/T', 'translation': 'ENSP00000288602', 'codons': 'cGa/cAa', 'id': 'rs730880415', 'seq_region_name': 'ENSP00000288602', 'end': 360, 'Parent': 'ENST00000288602', 'sift': 0.09}, {'polyphen': 0.994, 'type': 'missense_variant', 'residues': 'A/G', 'start': 537, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/C', 'codons': 'gCc/gGc', 'id': 'rs1370423184', 'seq_region_name': 'ENSP00000288602', 'end': 537, 'Parent': 'ENST00000288602', 'sift': 0}, {'type': 'missense_variant', 'polyphen': 0.121, 'residues': 'T/S', 'codons': 'Aca/Tca', 'allele': 'T/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 57, 'Parent': 'ENST00000288602', 'end': 57, 'id': 'rs1354935301', 'seq_region_name': 'ENSP00000288602', 'sift': 0.06}, {'type': 'missense_variant', 'polyphen': 0.656, 'residues': 'R/H', 'clinical_significance': ['likely pathogenic'], 'feature_type': 'transcript_variation', 'start': 260, 'codons': 'cGc/cAc', 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'end': 260, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1586213318', 'Parent': 'ENST00000288602', 'sift': 0.01}, {'start': 218, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'codons': 'Act/Gct', 'polyphen': 0.364, 'residues': 'T/A', 'type': 'missense_variant', 'sift': 0.74, 'seq_region_name': 'ENSP00000288602', 'id': 'rs766353093', 'end': 218, 'Parent': 'ENST00000288602'}, {'Parent': 'ENST00000288602', 'end': 27, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1247014863', 'sift': 0.4, 'polyphen': 0.069, 'type': 'missense_variant', 'residues': 'A/T', 'codons': 'Gcc/Acc', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 27}, {'sift': 0, 'id': 'rs121913337', 'seq_region_name': 'ENSP00000288602', 'end': 634, 'Parent': 'ENST00000288602', 'start': 634, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/C/T', 'codons': 'gaT/gaG/gaA', 'polyphen': 0.003, 'residues': 'D/E/E', 'type': 'missense_variant'}, {'polyphen': 0.199, 'type': 'missense_variant', 'residues': 'R/P/Q', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 403, 'codons': 'cGg/cCg/cAg', 'allele': 'C/G/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'end': 403, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1024852197', 'Parent': 'ENST00000288602', 'sift': 0.11}, {'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'cGg/cAg', 'start': 775, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'polyphen': 0.999, 'residues': 'R/Q', 'type': 'missense_variant', 'sift': 0.11, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1436193121', 'end': 775}, {'residues': 'I/V', 'polyphen': 0.036, 'type': 'missense_variant', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C', 'codons': 'Ata/Gta', 'start': 421, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1002249540', 'end': 421, 'sift': 0.52}, {'Parent': 'ENST00000288602', 'id': 'rs768694358', 'seq_region_name': 'ENSP00000288602', 'end': 171, 'sift': 0.02, 'polyphen': 0.562, 'residues': 'A/T', 'type': 'missense_variant', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'codons': 'Gca/Aca', 'start': 171, 'clinical_significance': [], 'feature_type': 'transcript_variation'}, {'Parent': 'ENST00000288602', 'id': 'rs1022449496', 'seq_region_name': 'ENSP00000288602', 'end': 447, 'sift': 0.12, 'type': 'missense_variant', 'polyphen': 1, 'residues': 'P/T', 'allele': 'G/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Cct/Act', 'start': 447, 'feature_type': 'transcript_variation', 'clinical_significance': []}, {'seq_region_name': 'ENSP00000288602', 'id': 'rs397507455', 'end': 14, 'Parent': 'ENST00000288602', 'sift': 0.21, 'residues': 'P/L/R', 'polyphen': 0, 'type': 'missense_variant', 'start': 14, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'allele': 'G/A/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'cCg/cTg/cGg'}, {'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 664, 'codons': 'Gtc/Atc', 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'type': 'missense_variant', 'polyphen': 0.44, 'residues': 'V/I', 'sift': 0, 'end': 664, 'seq_region_name': 'ENSP00000288602', 'id': 'rs781330698', 'Parent': 'ENST00000288602'}, {'Parent': 'ENST00000288602', 'end': 117, 'id': 'rs755475967', 'seq_region_name': 'ENSP00000288602', 'sift': 0.36, 'type': 'missense_variant', 'polyphen': 0.012, 'residues': 'M/V', 'codons': 'Atg/Gtg', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 117}, {'end': 771, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1562931107', 'Parent': 'ENST00000288602', 'sift': 0, 'polyphen': 0.999, 'type': 'missense_variant', 'residues': 'P/S', 'clinical_significance': ['likely pathogenic'], 'feature_type': 'transcript_variation', 'start': 771, 'codons': 'Ccc/Tcc', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/A'}, {'type': 'missense_variant', 'polyphen': 0.23, 'residues': 'G/A', 'start': 69, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'allele': 'C/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'gGt/gCt', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1554412417', 'end': 69, 'Parent': 'ENST00000288602', 'sift': 0.58}, {'polyphen': 0.985, 'residues': 'G/E', 'type': 'missense_variant', 'start': 495, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'codons': 'gGg/gAg', 'id': 'rs1343787672', 'seq_region_name': 'ENSP00000288602', 'end': 495, 'Parent': 'ENST00000288602', 'sift': 0.82}, {'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'codons': 'aGa/aAa', 'start': 432, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'residues': 'R/K', 'polyphen': 0.111, 'type': 'missense_variant', 'sift': 0.29, 'Parent': 'ENST00000288602', 'id': 'rs906595485', 'seq_region_name': 'ENSP00000288602', 'end': 432}, {'polyphen': 0.159, 'type': 'missense_variant', 'residues': 'L/V', 'start': 185, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/C', 'codons': 'Ctg/Gtg', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1354338741', 'end': 185, 'Parent': 'ENST00000288602', 'sift': 0.02}, {'id': 'rs1562985576', 'seq_region_name': 'ENSP00000288602', 'end': 94, 'Parent': 'ENST00000288602', 'sift': 0.56, 'polyphen': 0.03, 'type': 'missense_variant', 'residues': 'Q/R', 'start': 94, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'codons': 'cAa/cGa'}, {'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A/C/T', 'codons': 'cCc/cTc/cGc/cAc', 'start': 442, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'residues': 'P/L/R/H', 'polyphen': 1, 'type': 'missense_variant', 'sift': 0.08, 'Parent': 'ENST00000288602', 'id': 'rs199927105', 'seq_region_name': 'ENSP00000288602', 'end': 442}, {'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C', 'codons': 'gAt/gGt', 'start': 143, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'residues': 'D/G', 'polyphen': 0.101, 'type': 'missense_variant', 'sift': 0.1, 'Parent': 'ENST00000288602', 'id': 'rs763920732', 'seq_region_name': 'ENSP00000288602', 'end': 143}, {'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 686, 'codons': 'Ctg/Gtg', 'minor_allele_frequency': None, 'allele': 'G/C', 'translation': 'ENSP00000288602', 'polyphen': 0.596, 'type': 'missense_variant', 'residues': 'L/V', 'sift': 0, 'end': 686, 'id': 'rs758803218', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602'}, {'codons': 'Gtt/Ttt', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/A', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 175, 'residues': 'V/F', 'polyphen': 0.771, 'type': 'missense_variant', 'sift': 0.02, 'Parent': 'ENST00000288602', 'end': 175, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1554404622'}, {'sift': 0, 'Parent': 'ENST00000288602', 'end': 641, 'id': 'rs121913364', 'seq_region_name': 'ENSP00000288602', 'codons': 'Aaa/Gaa/Caa', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C/G', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'start': 641, 'polyphen': 0.345, 'type': 'missense_variant', 'residues': 'K/E/Q'}, {'residues': 'S/F', 'polyphen': 1, 'type': 'missense_variant', 'codons': 'tCt/tTt', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 439, 'Parent': 'ENST00000288602', 'end': 439, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1554400234', 'sift': 0.01}, {'end': 319, 'id': 'rs886041599', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.03, 'polyphen': 0.089, 'residues': 'S/F', 'type': 'missense_variant', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 319, 'codons': 'tCc/tTc', 'allele': 'G/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602'}, {'sift': 0.04, 'id': 'rs1472917101', 'seq_region_name': 'ENSP00000288602', 'end': 410, 'Parent': 'ENST00000288602', 'start': 410, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'allele': 'G/C', 'translation': 'ENSP00000288602', 'codons': 'cCc/cGc', 'polyphen': 0.998, 'type': 'missense_variant', 'residues': 'P/R'}, {'Parent': 'ENST00000288602', 'id': 'rs762305547', 'seq_region_name': 'ENSP00000288602', 'end': 322, 'sift': 1, 'residues': 'A/S/T', 'polyphen': 0, 'type': 'missense_variant', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/A/T', 'codons': 'Gcc/Tcc/Acc', 'start': 322, 'clinical_significance': [], 'feature_type': 'transcript_variation'}, {'polyphen': 0.04, 'type': 'missense_variant', 'residues': 'M/R', 'start': 117, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'A/C', 'codons': 'aTg/aGg', 'id': 'rs397507461', 'seq_region_name': 'ENSP00000288602', 'end': 117, 'Parent': 'ENST00000288602', 'sift': 0.55}, {'codons': 'gaT/gaA', 'allele': 'A/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 352, 'residues': 'D/E', 'polyphen': 0.999, 'type': 'missense_variant', 'sift': 0.33, 'Parent': 'ENST00000288602', 'end': 352, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1398817791'}, {'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs755214031', 'end': 612, 'sift': 0, 'type': 'missense_variant', 'polyphen': 0.656, 'residues': 'I/F/V', 'allele': 'T/A/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Atc/Ttc/Gtc', 'start': 612, 'feature_type': 'transcript_variation', 'clinical_significance': []}, {'seq_region_name': 'ENSP00000288602', 'id': 'rs551111837', 'end': 583, 'Parent': 'ENST00000288602', 'sift': 1, 'polyphen': 0.096, 'type': 'missense_variant', 'residues': 'I/V', 'start': 583, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'allele': 'T/C', 'translation': 'ENSP00000288602', 'codons': 'Atc/Gtc'}, {'Parent': 'ENST00000288602', 'end': 115, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1388495628', 'sift': 0.2, 'type': 'missense_variant', 'polyphen': 0.197, 'residues': 'A/V', 'codons': 'gCa/gTa', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/A', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 115}, {'sift': 0.24, 'Parent': 'ENST00000288602', 'end': 41, 'id': 'rs1186254108', 'seq_region_name': 'ENSP00000288602', 'codons': 'cCt/cTt', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/A', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 41, 'polyphen': 0.007, 'residues': 'P/L', 'type': 'missense_variant'}, {'polyphen': 0.007, 'residues': 'N/D/H', 'type': 'missense_variant', 'codons': 'Aat/Gat/Cat', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C/G', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'start': 621, 'Parent': 'ENST00000288602', 'end': 621, 'id': 'rs180177040', 'seq_region_name': 'ENSP00000288602', 'sift': 0.04}, {'allele': 'T/A/C', 'minor_allele_frequency': 0.0002, 'translation': 'ENSP00000288602', 'codons': 'Att/Ttt/Gtt', 'start': 754, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'residues': 'I/F/V', 'polyphen': 0.412, 'type': 'missense_variant', 'sift': 0, 'Parent': 'ENST00000288602', 'id': 'rs555976452', 'seq_region_name': 'ENSP00000288602', 'end': 754}, {'sift': 0, 'Parent': 'ENST00000288602', 'id': 'rs180177037', 'seq_region_name': 'ENSP00000288602', 'end': 539, 'minor_allele_frequency': None, 'allele': 'T/C/G', 'translation': 'ENSP00000288602', 'codons': 'Aaa/Gaa/Caa', 'start': 539, 'clinical_significance': ['uncertain significance', 'likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'polyphen': 0.427, 'type': 'missense_variant', 'residues': 'K/E/Q'}, {'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs749287111', 'end': 629, 'sift': 0, 'polyphen': 0.162, 'type': 'missense_variant', 'residues': 'T/A/P', 'minor_allele_frequency': None, 'allele': 'T/C/G', 'translation': 'ENSP00000288602', 'codons': 'Aca/Gca/Cca', 'start': 629, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic']}, {'sift': 0.2, 'Parent': 'ENST00000288602', 'id': 'rs1183933401', 'seq_region_name': 'ENSP00000288602', 'end': 411, 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'codons': 'Agt/Ggt', 'start': 411, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'polyphen': 0.12, 'type': 'missense_variant', 'residues': 'S/G'}, {'allele': 'G/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'cCa/cGa', 'start': 299, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'polyphen': 1, 'type': 'missense_variant', 'residues': 'P/R', 'sift': 0.01, 'Parent': 'ENST00000288602', 'id': 'rs1259806158', 'seq_region_name': 'ENSP00000288602', 'end': 299}, {'sift': 0, 'end': 20, 'id': 'rs1321934224', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 20, 'codons': 'Aac/Gac', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C', 'polyphen': 0.992, 'residues': 'N/D', 'type': 'missense_variant'}, {'codons': 'aTC/aGT', 'minor_allele_frequency': None, 'allele': 'GA/AC', 'translation': 'ENSP00000288602', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 795, 'type': 'missense_variant', 'polyphen': 0.999, 'residues': 'I/S', 'sift': 0.08, 'Parent': 'ENST00000288602', 'end': 795, 'id': 'rs1064795542', 'seq_region_name': 'ENSP00000288602'}, {'Parent': 'ENST00000288602', 'end': 435, 'seq_region_name': 'ENSP00000288602', 'id': 'rs397516886', 'sift': 0.5, 'type': 'missense_variant', 'polyphen': 0.015, 'residues': 'T/A', 'codons': 'Acc/Gcc', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 435}, {'polyphen': 0.835, 'type': 'missense_variant', 'residues': 'D/H/N', 'start': 634, 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/G/T', 'codons': 'Gat/Cat/Aat', 'seq_region_name': 'ENSP00000288602', 'id': 'rs397516896', 'end': 634, 'Parent': 'ENST00000288602', 'sift': 0}, {'end': 389, 'seq_region_name': 'ENSP00000288602', 'id': 'rs577372072', 'Parent': 'ENST00000288602', 'sift': 0.13, 'polyphen': 0.799, 'residues': 'R/H', 'type': 'missense_variant', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 389, 'codons': 'cGt/cAt', 'translation': 'ENSP00000288602', 'minor_allele_frequency': 0.0002, 'allele': 'C/T'}, {'codons': 'Aga/Gga', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 483, 'polyphen': 0.966, 'residues': 'R/G', 'type': 'missense_variant', 'sift': 0.2, 'Parent': 'ENST00000288602', 'end': 483, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1256277078'}, {'sift': 0, 'end': 634, 'id': 'rs121913338', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance', 'likely pathogenic', 'pathogenic'], 'start': 634, 'codons': 'gAt/gTt/gGt/gCt', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/A/C/G', 'polyphen': 0.569, 'residues': 'D/V/G/A', 'type': 'missense_variant'}, {'clinical_significance': ['uncertain significance', 'likely pathogenic'], 'feature_type': 'transcript_variation', 'start': 511, 'codons': 'Gtc/Ttc/Atc', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/A/T', 'polyphen': 0.932, 'residues': 'V/F/I', 'type': 'missense_variant', 'sift': 0.01, 'end': 511, 'seq_region_name': 'ENSP00000288602', 'id': 'rs121913376', 'Parent': 'ENST00000288602'}, {'sift': 0, 'end': 687, 'id': 'rs1562939198', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 687, 'codons': 'tAt/tGt', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'polyphen': 0.993, 'type': 'missense_variant', 'residues': 'Y/C'}, {'sift': 0.03, 'Parent': 'ENST00000288602', 'end': 722, 'id': 'rs776664982', 'seq_region_name': 'ENSP00000288602', 'codons': 'Cgg/Tgg', 'minor_allele_frequency': None, 'allele': 'G/A', 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 722, 'polyphen': 0.912, 'residues': 'R/W', 'type': 'missense_variant'}, {'start': 574, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'allele': 'C/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'gGc/gTc', 'type': 'missense_variant', 'polyphen': 1, 'residues': 'G/V', 'sift': 0, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1442683518', 'end': 574, 'Parent': 'ENST00000288602'}, {'Parent': 'ENST00000288602', 'end': 431, 'id': 'rs13230015', 'seq_region_name': 'ENSP00000288602', 'sift': 0.72, 'polyphen': 0.017, 'residues': 'F/L', 'type': 'missense_variant', 'codons': 'Ttc/Ctc', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/G', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 431}, {'start': 34, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely benign'], 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/A', 'codons': 'gCg/gTg', 'polyphen': 0.005, 'residues': 'A/V', 'type': 'missense_variant', 'sift': 0.25, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1424449802', 'end': 34, 'Parent': 'ENST00000288602'}, {'allele': 'A/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'aaT/aaA', 'start': 621, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'residues': 'N/K', 'polyphen': 0.822, 'type': 'missense_variant', 'sift': 0, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs397516895', 'end': 621}, {'id': 'rs121913362', 'seq_region_name': 'ENSP00000288602', 'end': 632, 'Parent': 'ENST00000288602', 'sift': 0, 'type': 'missense_variant', 'polyphen': 0.045, 'residues': 'I/M', 'start': 632, 'clinical_significance': ['likely pathogenic'], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'allele': 'T/C', 'translation': 'ENSP00000288602', 'codons': 'atA/atG'}, {'residues': 'A/G', 'polyphen': 0.007, 'type': 'missense_variant', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/C', 'codons': 'gCt/gGt', 'start': 776, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'id': 'rs1348443729', 'seq_region_name': 'ENSP00000288602', 'end': 776, 'sift': 0.01}, {'sift': 0.04, 'seq_region_name': 'ENSP00000288602', 'id': 'rs397507469', 'end': 257, 'Parent': 'ENST00000288602', 'start': 257, 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'allele': 'G/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Cag/Aag', 'polyphen': 0.024, 'residues': 'Q/K', 'type': 'missense_variant'}, {'start': 792, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'codons': 'Aaa/Gaa', 'polyphen': 0.843, 'type': 'missense_variant', 'residues': 'K/E', 'sift': 0, 'id': 'rs752026903', 'seq_region_name': 'ENSP00000288602', 'end': 792, 'Parent': 'ENST00000288602'}, {'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 138, 'codons': 'tTt/tCt', 'allele': 'A/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'residues': 'F/S', 'polyphen': 0.096, 'type': 'missense_variant', 'sift': 0.43, 'end': 138, 'id': 'rs1339350570', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602'}, {'sift': 0, 'Parent': 'ENST00000288602', 'id': 'rs746010267', 'seq_region_name': 'ENSP00000288602', 'end': 673, 'allele': 'A/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Tac/Cac', 'start': 673, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'residues': 'Y/H', 'polyphen': 1, 'type': 'missense_variant'}, {'sift': 0.17, 'id': 'rs1273585752', 'seq_region_name': 'ENSP00000288602', 'end': 30, 'Parent': 'ENST00000288602', 'start': 30, 'clinical_significance': ['likely benign'], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'allele': 'C/T', 'translation': 'ENSP00000288602', 'codons': 'gGc/gAc', 'type': 'missense_variant', 'polyphen': 0.433, 'residues': 'G/D'}, {'sift': 0.04, 'Parent': 'ENST00000288602', 'id': 'rs774666543', 'seq_region_name': 'ENSP00000288602', 'end': 169, 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Gta/Ata', 'start': 169, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'residues': 'V/I', 'polyphen': 0.998, 'type': 'missense_variant'}, {'start': 498, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'codons': 'Aca/Gca', 'type': 'missense_variant', 'polyphen': 0.699, 'residues': 'T/A', 'sift': 0.5, 'id': 'rs1461539805', 'seq_region_name': 'ENSP00000288602', 'end': 498, 'Parent': 'ENST00000288602'}, {'end': 300, 'id': 'rs372239545', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.44, 'type': 'missense_variant', 'polyphen': 0, 'residues': 'I/V', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 300, 'codons': 'Ata/Gta', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C'}, {'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 389, 'codons': 'Cgt/Tgt/Agt', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A/T', 'polyphen': 0.021, 'residues': 'R/C/S', 'type': 'missense_variant', 'sift': 0.05, 'end': 389, 'id': 'rs397507472', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602'}, {'codons': 'Gta/Ata', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 238, 'polyphen': 0.073, 'residues': 'V/I', 'type': 'missense_variant', 'sift': 0.16, 'Parent': 'ENST00000288602', 'end': 238, 'seq_region_name': 'ENSP00000288602', 'id': 'rs373935661'}, {'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1554389860', 'end': 716, 'sift': 0, 'polyphen': 1, 'residues': 'P/S', 'type': 'missense_variant', 'minor_allele_frequency': None, 'allele': 'G/A', 'translation': 'ENSP00000288602', 'codons': 'Cca/Tca', 'start': 716, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance']}, {'sift': 0.25, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1176574782', 'end': 125, 'allele': 'A/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Tct/Cct', 'start': 125, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'residues': 'S/P', 'polyphen': 0.015, 'type': 'missense_variant'}, {'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic'], 'start': 247, 'codons': 'tTt/tCt', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/G', 'type': 'missense_variant', 'polyphen': 0.877, 'residues': 'F/S', 'sift': 0, 'end': 247, 'seq_region_name': 'ENSP00000288602', 'id': 'rs397507467', 'Parent': 'ENST00000288602'}, {'Parent': 'ENST00000288602', 'id': 'rs768303459', 'seq_region_name': 'ENSP00000288602', 'end': 588, 'sift': 0.09, 'residues': 'F/L', 'polyphen': 0.999, 'type': 'missense_variant', 'minor_allele_frequency': None, 'allele': 'A/G', 'translation': 'ENSP00000288602', 'codons': 'Ttt/Ctt', 'start': 588, 'clinical_significance': [], 'feature_type': 'transcript_variation'}, {'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 728, 'codons': 'gCc/gTc', 'allele': 'G/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'polyphen': 0.192, 'residues': 'A/V', 'type': 'missense_variant', 'sift': 0, 'end': 728, 'id': 'rs1442061595', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602'}, {'seq_region_name': 'ENSP00000288602', 'id': 'rs201481342', 'end': 342, 'Parent': 'ENST00000288602', 'sift': 0.35, 'polyphen': 0.018, 'type': 'missense_variant', 'residues': 'I/V', 'start': 342, 'clinical_significance': ['uncertain significance', 'likely benign'], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': 0.0002, 'allele': 'T/C', 'codons': 'Att/Gtt'}, {'codons': 'gGa/gTa/gCa/gAa', 'minor_allele_frequency': None, 'allele': 'C/A/G/T', 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'start': 504, 'type': 'missense_variant', 'polyphen': 0.988, 'residues': 'G/V/A/E', 'sift': 0, 'Parent': 'ENST00000288602', 'end': 504, 'id': 'rs121913348', 'seq_region_name': 'ENSP00000288602'}, {'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'GC/TT', 'codons': 'GCa/AAa', 'start': 521, 'clinical_significance': ['likely pathogenic'], 'feature_type': 'transcript_variation', 'polyphen': 1, 'type': 'missense_variant', 'residues': 'A/K', 'sift': 0, 'Parent': 'ENST00000288602', 'id': 'rs1586126644', 'seq_region_name': 'ENSP00000288602', 'end': 521}, {'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'cGa/cAa', 'start': 477, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely benign'], 'type': 'missense_variant', 'polyphen': 0.723, 'residues': 'R/Q', 'sift': 0.1, 'Parent': 'ENST00000288602', 'id': 'rs998233805', 'seq_region_name': 'ENSP00000288602', 'end': 477}, {'sift': 0.01, 'seq_region_name': 'ENSP00000288602', 'id': 'rs765883358', 'end': 86, 'Parent': 'ENST00000288602', 'start': 86, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'minor_allele_frequency': None, 'allele': 'T/A', 'translation': 'ENSP00000288602', 'codons': 'Acc/Tcc', 'polyphen': 0.807, 'type': 'missense_variant', 'residues': 'T/S'}, {'sift': 0, 'Parent': 'ENST00000288602', 'end': 114, 'id': 'rs1174578809', 'seq_region_name': 'ENSP00000288602', 'codons': 'tCt/tGt', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/C', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 114, 'type': 'missense_variant', 'polyphen': 0.919, 'residues': 'S/C'}, {'sift': 0.56, 'seq_region_name': 'ENSP00000288602', 'id': 'rs749140487', 'end': 257, 'Parent': 'ENST00000288602', 'start': 257, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'minor_allele_frequency': None, 'allele': 'C/G', 'translation': 'ENSP00000288602', 'codons': 'caG/caC', 'polyphen': 0.888, 'residues': 'Q/H', 'type': 'missense_variant'}, {'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1052947619', 'end': 126, 'sift': 0.01, 'residues': 'S/F', 'polyphen': 0.337, 'type': 'missense_variant', 'allele': 'G/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'tCt/tTt', 'start': 126, 'feature_type': 'transcript_variation', 'clinical_significance': []}, {'polyphen': 0.936, 'residues': 'S/P', 'type': 'missense_variant', 'codons': 'Tca/Cca', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'A/G', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 434, 'Parent': 'ENST00000288602', 'end': 434, 'id': 'rs923739321', 'seq_region_name': 'ENSP00000288602', 'sift': 0.03}, {'residues': 'D/V', 'polyphen': 0.999, 'type': 'missense_variant', 'allele': 'T/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'gAc/gTc', 'start': 22, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1414371670', 'end': 22, 'sift': 0}, {'start': 288, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'allele': 'A/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'tTg/tCg', 'polyphen': 0.767, 'residues': 'L/S', 'type': 'missense_variant', 'sift': 0.61, 'id': 'rs768365723', 'seq_region_name': 'ENSP00000288602', 'end': 288, 'Parent': 'ENST00000288602'}, {'sift': 0.75, 'id': 'rs1562954612', 'seq_region_name': 'ENSP00000288602', 'end': 584, 'Parent': 'ENST00000288602', 'start': 584, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'allele': 'A/G', 'translation': 'ENSP00000288602', 'codons': 'aTt/aCt', 'type': 'missense_variant', 'polyphen': 0.754, 'residues': 'I/T'}, {'sift': 0.08, 'end': 337, 'id': 'rs200373902', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 337, 'codons': 'Tca/Cca', 'allele': 'A/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'polyphen': 0.059, 'type': 'missense_variant', 'residues': 'S/P'}, {'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1222192591', 'end': 18, 'sift': 0.25, 'type': 'missense_variant', 'polyphen': 0.992, 'residues': 'L/V', 'allele': 'G/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Ctg/Gtg', 'start': 18, 'feature_type': 'transcript_variation', 'clinical_significance': []}, {'sift': 0.23, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs879059895', 'end': 792, 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C', 'codons': 'aAa/aGa', 'start': 792, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'type': 'missense_variant', 'polyphen': 0.159, 'residues': 'K/R'}, {'polyphen': 1, 'residues': 'L/R/P/Q', 'type': 'missense_variant', 'codons': 'cTg/cGg/cCg/cAg', 'allele': 'A/C/G/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'start': 565, 'Parent': 'ENST00000288602', 'end': 565, 'seq_region_name': 'ENSP00000288602', 'id': 'rs869025340', 'sift': 0}, {'sift': 0.64, 'Parent': 'ENST00000288602', 'end': 326, 'id': 'rs775040765', 'seq_region_name': 'ENSP00000288602', 'codons': 'Att/Gtt', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'feature_type': 'transcript_variation', 'clinical_significance': ['benign'], 'start': 326, 'polyphen': 0, 'type': 'missense_variant', 'residues': 'I/V'}, {'start': 403, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'minor_allele_frequency': None, 'allele': 'G/A/C', 'translation': 'ENSP00000288602', 'codons': 'Cgg/Tgg/Ggg', 'polyphen': 0.551, 'residues': 'R/W/G', 'type': 'missense_variant', 'sift': 0.01, 'id': 'rs906437759', 'seq_region_name': 'ENSP00000288602', 'end': 403, 'Parent': 'ENST00000288602'}, {'start': 649, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'allele': 'C/A', 'translation': 'ENSP00000288602', 'codons': 'caG/caT', 'polyphen': 0.399, 'type': 'missense_variant', 'residues': 'Q/H', 'sift': 0, 'seq_region_name': 'ENSP00000288602', 'id': 'rs746269606', 'end': 649, 'Parent': 'ENST00000288602'}, {'polyphen': 1, 'type': 'missense_variant', 'residues': 'P/S', 'start': 343, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'allele': 'G/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Cca/Tca', 'id': 'rs758935249', 'seq_region_name': 'ENSP00000288602', 'end': 343, 'Parent': 'ENST00000288602', 'sift': 0.18}, {'Parent': 'ENST00000288602', 'id': 'rs1008100536', 'seq_region_name': 'ENSP00000288602', 'end': 497, 'sift': 0, 'polyphen': 0.984, 'residues': 'I/F', 'type': 'missense_variant', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/A', 'codons': 'Att/Ttt', 'start': 497, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance']}, {'polyphen': 0.998, 'type': 'missense_variant', 'residues': 'S/G', 'codons': 'Agt/Ggt', 'minor_allele_frequency': None, 'allele': 'T/C', 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 487, 'Parent': 'ENST00000288602', 'end': 487, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1341682230', 'sift': 0.24}, {'sift': None, 'Parent': 'ENST00000288602', 'end': 143, 'id': 'rs370069295', 'seq_region_name': 'ENSP00000288602', 'codons': 'gaT/gaC/gaA', 'allele': 'A/G/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 143, 'polyphen': None, 'type': 'missense_variant', 'residues': 'D/D/E'}, {'start': 510, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic'], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/G', 'codons': 'Aca/Cca', 'residues': 'T/P', 'polyphen': 1, 'type': 'missense_variant', 'sift': 0, 'id': 'rs1586140512', 'seq_region_name': 'ENSP00000288602', 'end': 510, 'Parent': 'ENST00000288602'}, {'end': 245, 'id': 'rs397507466', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0, 'type': 'missense_variant', 'polyphen': 1, 'residues': 'L/F/L/F', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'start': 245, 'codons': 'ttA/ttT/ttG/ttC', 'minor_allele_frequency': None, 'allele': 'T/A/C/G', 'translation': 'ENSP00000288602'}, {'sift': 1, 'Parent': 'ENST00000288602', 'id': 'rs1562957691', 'seq_region_name': 'ENSP00000288602', 'end': 475, 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'codons': 'aGg/aAg', 'start': 475, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'type': 'missense_variant', 'polyphen': 0.024, 'residues': 'R/K'}, {'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 800, 'codons': 'tAt/tGt', 'allele': 'T/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'polyphen': 0.993, 'residues': 'Y/C', 'type': 'missense_variant', 'sift': 0, 'end': 800, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1305119093', 'Parent': 'ENST00000288602'}, {'sift': 0.01, 'end': 408, 'id': 'rs1475054572', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 408, 'codons': 'Cgg/Tgg', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'polyphen': 0.858, 'residues': 'R/W', 'type': 'missense_variant'}, {'sift': 0.01, 'end': 366, 'id': 'rs1038048880', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic'], 'start': 366, 'codons': 'Gct/Cct/Act', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/G/T', 'residues': 'A/P/T', 'polyphen': 0.994, 'type': 'missense_variant'}, {'polyphen': 0.936, 'type': 'missense_variant', 'residues': 'A/P', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 797, 'codons': 'Gca/Cca', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/G', 'end': 797, 'id': 'rs1220285993', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.16}, {'type': 'missense_variant', 'polyphen': 0.074, 'residues': 'S/F/C', 'minor_allele_frequency': None, 'allele': 'G/A/C', 'translation': 'ENSP00000288602', 'codons': 'tCc/tTc/tGc', 'start': 317, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs779101864', 'end': 317, 'sift': 0.69}, {'sift': 0.02, 'Parent': 'ENST00000288602', 'end': 503, 'id': 'rs1562957000', 'seq_region_name': 'ENSP00000288602', 'codons': 'Att/Gtt', 'minor_allele_frequency': None, 'allele': 'T/C', 'translation': 'ENSP00000288602', 'clinical_significance': ['likely pathogenic'], 'feature_type': 'transcript_variation', 'start': 503, 'type': 'missense_variant', 'polyphen': 0.086, 'residues': 'I/V'}, {'start': 523, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic'], 'allele': 'T/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'aaA/aaC', 'polyphen': 1, 'residues': 'K/N', 'type': 'missense_variant', 'sift': 0, 'id': 'rs727504375', 'seq_region_name': 'ENSP00000288602', 'end': 523, 'Parent': 'ENST00000288602'}, {'polyphen': 0.054, 'residues': 'F/I', 'type': 'missense_variant', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 109, 'codons': 'Ttt/Att', 'allele': 'A/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'end': 109, 'id': 'rs397507460', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.24}, {'seq_region_name': 'ENSP00000288602', 'id': 'rs537281594', 'end': 400, 'Parent': 'ENST00000288602', 'sift': 0.05, 'residues': 'R/C', 'polyphen': 0.734, 'type': 'missense_variant', 'start': 400, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'allele': 'G/A', 'minor_allele_frequency': 0.0002, 'translation': 'ENSP00000288602', 'codons': 'Cgc/Tgc'}, {'allele': 'C/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'gGt/gTt', 'start': 8, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'polyphen': 1, 'residues': 'G/V', 'type': 'missense_variant', 'sift': 0.02, 'Parent': 'ENST00000288602', 'id': 'rs774925605', 'seq_region_name': 'ENSP00000288602', 'end': 8}, {'codons': 'Gat/Aat', 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 143, 'polyphen': 0.944, 'type': 'missense_variant', 'residues': 'D/N', 'sift': 0.08, 'Parent': 'ENST00000288602', 'end': 143, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1586339585'}, {'Parent': 'ENST00000288602', 'end': 148, 'seq_region_name': 'ENSP00000288602', 'id': 'rs773266363', 'sift': 0.03, 'type': 'missense_variant', 'polyphen': 0.97, 'residues': 'N/I', 'codons': 'aAc/aTc', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/A', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 148}, {'polyphen': 0.135, 'type': 'missense_variant', 'residues': 'R/Q', 'codons': 'cGa/cAa', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 643, 'Parent': 'ENST00000288602', 'end': 643, 'id': 'rs372569965', 'seq_region_name': 'ENSP00000288602', 'sift': 0}, {'sift': 0.01, 'Parent': 'ENST00000288602', 'end': 536, 'id': 'rs1554399258', 'seq_region_name': 'ENSP00000288602', 'codons': 'caA/caC', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/G', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 536, 'polyphen': 0.765, 'residues': 'Q/H', 'type': 'missense_variant'}, {'seq_region_name': 'ENSP00000288602', 'id': 'rs730880412', 'end': 25, 'Parent': 'ENST00000288602', 'sift': 0.59, 'type': 'missense_variant', 'polyphen': 0.001, 'residues': 'P/S/T', 'start': 25, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance', 'likely benign'], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A/T', 'codons': 'Ccc/Tcc/Acc'}, {'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'codons': 'Atg/Gtg', 'start': 729, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'polyphen': 0.116, 'type': 'missense_variant', 'residues': 'M/V', 'sift': 0.03, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1028549781', 'end': 729}, {'sift': 0, 'seq_region_name': 'ENSP00000288602', 'id': 'rs55939351', 'end': 632, 'Parent': 'ENST00000288602', 'start': 632, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'allele': 'A/G/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'aTa/aCa/aAa', 'polyphen': 0.308, 'residues': 'I/T/K', 'type': 'missense_variant'}, {'sift': 0, 'end': 11, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1225976306', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 11, 'codons': 'Ggc/Agc', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'polyphen': 0.006, 'type': 'missense_variant', 'residues': 'G/S'}, {'sift': 0.02, 'end': 325, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1238788540', 'Parent': 'ENST00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 325, 'codons': 'tCt/tGt', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/C', 'residues': 'S/C', 'polyphen': 0.547, 'type': 'missense_variant'}, {'sift': 0.21, 'id': 'rs1356557681', 'seq_region_name': 'ENSP00000288602', 'end': 10, 'Parent': 'ENST00000288602', 'start': 10, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'codons': 'Ggc/Agc', 'residues': 'G/S', 'polyphen': 0.119, 'type': 'missense_variant'}, {'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 348, 'codons': 'Cca/Gca', 'allele': 'G/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'type': 'missense_variant', 'polyphen': 0.015, 'residues': 'P/A', 'sift': 0.1, 'end': 348, 'id': 'rs766062968', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602'}, {'type': 'missense_variant', 'polyphen': 0.094, 'residues': 'A/S/T', 'allele': 'C/A/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Gcc/Tcc/Acc', 'start': 752, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic'], 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1481562268', 'end': 752, 'sift': 0.34}, {'end': 36, 'id': 'rs886041827', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.64, 'type': 'missense_variant', 'polyphen': 0.003, 'residues': 'S/F/C', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 36, 'codons': 'tCt/tTt/tGt', 'allele': 'G/A/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602'}, {'sift': 0, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs113488022', 'end': 640, 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/C/G/T', 'codons': 'gTg/gGg/gCg/gAg', 'start': 640, 'clinical_significance': ['likely pathogenic', 'pathogenic', 'drug response', 'other'], 'feature_type': 'transcript_variation', 'type': 'missense_variant', 'polyphen': 0.971, 'residues': 'V/G/A/E'}, {'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 44, 'codons': 'Ccg/Acg', 'minor_allele_frequency': None, 'allele': 'G/T', 'translation': 'ENSP00000288602', 'residues': 'P/T', 'polyphen': 0.381, 'type': 'missense_variant', 'sift': 0.19, 'end': 44, 'id': 'rs1279627804', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602'}, {'Parent': 'ENST00000288602', 'id': 'rs180177038', 'seq_region_name': 'ENSP00000288602', 'end': 541, 'sift': 0, 'polyphen': 0.973, 'type': 'missense_variant', 'residues': 'E/Q/K', 'allele': 'C/G/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Gaa/Caa/Aaa', 'start': 541, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic']}, {'residues': 'Q/R', 'polyphen': 0.999, 'type': 'missense_variant', 'start': 599, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C', 'codons': 'cAg/cGg', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1232455781', 'end': 599, 'Parent': 'ENST00000288602', 'sift': 0}, {'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'codons': 'cCt/cTt', 'start': 394, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'residues': 'P/L', 'polyphen': 0.145, 'type': 'missense_variant', 'sift': 0.06, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs987622439', 'end': 394}, {'polyphen': None, 'residues': 'L/L/V', 'type': 'missense_variant', 'codons': 'Cta/Tta/Gta', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A/C', 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'start': 637, 'Parent': 'ENST00000288602', 'end': 637, 'id': 'rs121913369', 'seq_region_name': 'ENSP00000288602', 'sift': None}, {'sift': 0.19, 'Parent': 'ENST00000288602', 'end': 113, 'id': 'rs747047638', 'seq_region_name': 'ENSP00000288602', 'codons': 'aGc/aAc', 'minor_allele_frequency': None, 'allele': 'C/T', 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 113, 'polyphen': 0.024, 'type': 'missense_variant', 'residues': 'S/N'}, {'Parent': 'ENST00000288602', 'end': 565, 'seq_region_name': 'ENSP00000288602', 'id': 'rs766837571', 'sift': 0.02, 'polyphen': 0.998, 'residues': 'L/V', 'type': 'missense_variant', 'codons': 'Ctg/Gtg', 'allele': 'G/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 565}, {'sift': 1, 'Parent': 'ENST00000288602', 'id': 'rs1333413472', 'seq_region_name': 'ENSP00000288602', 'end': 16, 'allele': 'G/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Cag/Gag', 'start': 16, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'polyphen': 0.082, 'type': 'missense_variant', 'residues': 'Q/E'}, {'type': 'missense_variant', 'polyphen': 0, 'residues': 'I/T', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/G', 'codons': 'aTt/aCt', 'start': 326, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs368435578', 'end': 326, 'sift': 0.73}, {'end': 61, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1265063696', 'Parent': 'ENST00000288602', 'sift': 0.05, 'residues': 'I/V', 'polyphen': 0.018, 'type': 'missense_variant', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 61, 'codons': 'Ata/Gta', 'minor_allele_frequency': None, 'allele': 'T/C', 'translation': 'ENSP00000288602'}, {'sift': 0, 'Parent': 'ENST00000288602', 'id': 'rs397507474', 'seq_region_name': 'ENSP00000288602', 'end': 523, 'minor_allele_frequency': None, 'allele': 'T/C/G', 'translation': 'ENSP00000288602', 'codons': 'Aaa/Gaa/Caa', 'start': 523, 'clinical_significance': ['likely pathogenic'], 'feature_type': 'transcript_variation', 'residues': 'K/E/Q', 'polyphen': 0.999, 'type': 'missense_variant'}, {'seq_region_name': 'ENSP00000288602', 'id': 'rs956815092', 'end': 134, 'Parent': 'ENST00000288602', 'sift': 0.16, 'polyphen': 0.522, 'type': 'missense_variant', 'residues': 'S/P', 'start': 134, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'allele': 'A/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Tct/Cct'}, {'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs780556975', 'end': 176, 'sift': 0, 'residues': 'T/I/R', 'polyphen': 0.599, 'type': 'missense_variant', 'minor_allele_frequency': None, 'allele': 'G/A/C', 'translation': 'ENSP00000288602', 'codons': 'aCa/aTa/aGa', 'start': 176, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance']}, {'codons': 'Ggc/Agc', 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 28, 'type': 'missense_variant', 'polyphen': 0.039, 'residues': 'G/S', 'sift': 0.71, 'Parent': 'ENST00000288602', 'end': 28, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1437317949'}, {'seq_region_name': 'ENSP00000288602', 'id': 'rs202209106', 'end': 350, 'Parent': 'ENST00000288602', 'sift': 0.09, 'polyphen': 0.945, 'residues': 'D/Y', 'type': 'missense_variant', 'start': 350, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/A', 'codons': 'Gat/Tat'}, {'end': 367, 'seq_region_name': 'ENSP00000288602', 'id': 'rs267601317', 'Parent': 'ENST00000288602', 'sift': 0, 'polyphen': 1, 'residues': 'P/L', 'type': 'missense_variant', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 367, 'codons': 'cCc/cTc', 'minor_allele_frequency': None, 'allele': 'G/A', 'translation': 'ENSP00000288602'}, {'codons': 'aGa/aAa', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 384, 'polyphen': 0.287, 'type': 'missense_variant', 'residues': 'R/K', 'sift': 0.25, 'Parent': 'ENST00000288602', 'end': 384, 'id': 'rs1264766156', 'seq_region_name': 'ENSP00000288602'}, {'id': 'rs1452646888', 'seq_region_name': 'ENSP00000288602', 'end': 463, 'Parent': 'ENST00000288602', 'sift': 0.21, 'polyphen': 0.021, 'type': 'missense_variant', 'residues': 'Q/H', 'start': 463, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'minor_allele_frequency': None, 'allele': 'C/G', 'translation': 'ENSP00000288602', 'codons': 'caG/caC'}, {'start': 120, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'codons': 'Gtt/Att', 'residues': 'V/I', 'polyphen': 0.146, 'type': 'missense_variant', 'sift': 0.18, 'seq_region_name': 'ENSP00000288602', 'id': 'rs766748977', 'end': 120, 'Parent': 'ENST00000288602'}, {'polyphen': 1, 'residues': 'G/C', 'type': 'missense_variant', 'codons': 'Ggt/Tgt', 'minor_allele_frequency': None, 'allele': 'C/A', 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 8, 'Parent': 'ENST00000288602', 'end': 8, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1205861530', 'sift': 0.01}, {'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance', 'likely pathogenic'], 'start': 626, 'codons': 'Gaa/Aaa', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'type': 'missense_variant', 'polyphen': 0.189, 'residues': 'E/K', 'sift': 0, 'end': 626, 'seq_region_name': 'ENSP00000288602', 'id': 'rs121913340', 'Parent': 'ENST00000288602'}, {'sift': 0, 'Parent': 'ENST00000288602', 'end': 765, 'id': 'rs1585931494', 'seq_region_name': 'ENSP00000288602', 'codons': 'Cac/Tac', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/A', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 765, 'residues': 'H/Y', 'polyphen': 0.998, 'type': 'missense_variant'}, {'sift': 0.02, 'end': 509, 'seq_region_name': 'ENSP00000288602', 'id': 'rs121913358', 'Parent': 'ENST00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 509, 'codons': 'GGA/AGT/AGC', 'minor_allele_frequency': None, 'allele': 'TCC/ACT/GCT', 'translation': 'ENSP00000288602', 'polyphen': 0.98, 'type': 'missense_variant', 'residues': 'G/S/S'}, {'polyphen': 0.792, 'type': 'missense_variant', 'residues': 'P/A', 'codons': 'Cca/Gca', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/C', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 341, 'Parent': 'ENST00000288602', 'end': 341, 'seq_region_name': 'ENSP00000288602', 'id': 'rs757653653', 'sift': 0.04}, {'codons': 'Cag/Gag', 'minor_allele_frequency': None, 'allele': 'G/C', 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 704, 'residues': 'Q/E', 'polyphen': 0.981, 'type': 'missense_variant', 'sift': 0, 'Parent': 'ENST00000288602', 'end': 704, 'seq_region_name': 'ENSP00000288602', 'id': 'rs754481830'}, {'codons': 'aTt/aCt', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/G', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 750, 'polyphen': 0.622, 'type': 'missense_variant', 'residues': 'I/T', 'sift': 0, 'Parent': 'ENST00000288602', 'end': 750, 'id': 'rs1375912300', 'seq_region_name': 'ENSP00000288602'}, {'codons': 'ggTGat/ggGAat', 'minor_allele_frequency': None, 'allele': 'CA/TC', 'translation': 'ENSP00000288602', 'clinical_significance': ['pathogenic'], 'feature_type': 'transcript_variation', 'start': 633, 'type': 'missense_variant', 'polyphen': None, 'residues': 'GD/GN', 'sift': None, 'Parent': 'ENST00000288602', 'end': 634, 'id': 'rs1057519718', 'seq_region_name': 'ENSP00000288602'}, {'sift': 0.2, 'end': 384, 'id': 'rs545495379', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['benign'], 'start': 384, 'codons': 'Aga/Gga', 'allele': 'T/C', 'minor_allele_frequency': 0.0002, 'translation': 'ENSP00000288602', 'residues': 'R/G', 'polyphen': 0.018, 'type': 'missense_variant'}, {'sift': 0.05, 'Parent': 'ENST00000288602', 'end': 16, 'id': 'rs1563042542', 'seq_region_name': 'ENSP00000288602', 'codons': 'caG/caC', 'minor_allele_frequency': None, 'allele': 'C/G', 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 16, 'polyphen': 0.57, 'residues': 'Q/H', 'type': 'missense_variant'}, {'sift': 0, 'id': 'rs397507478', 'seq_region_name': 'ENSP00000288602', 'end': 571, 'Parent': 'ENST00000288602', 'start': 571, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'minor_allele_frequency': None, 'allele': 'C/A/G', 'translation': 'ENSP00000288602', 'codons': 'tGg/tTg/tCg', 'polyphen': 0.999, 'residues': 'W/L/S', 'type': 'missense_variant'}, {'sift': 0.31, 'Parent': 'ENST00000288602', 'end': 802, 'seq_region_name': 'ENSP00000288602', 'id': 'rs775889922', 'codons': 'gCg/gTg', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/A', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 802, 'polyphen': 0.001, 'type': 'missense_variant', 'residues': 'A/V'}, {'end': 81, 'id': 'rs1371538157', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.04, 'type': 'missense_variant', 'polyphen': 0.999, 'residues': 'A/V', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 81, 'codons': 'gCc/gTc', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/A'}, {'residues': 'G/R', 'polyphen': 0.988, 'type': 'missense_variant', 'start': 106, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'codons': 'Gga/Aga', 'id': 'rs749247588', 'seq_region_name': 'ENSP00000288602', 'end': 106, 'Parent': 'ENST00000288602', 'sift': 0}, {'residues': 'A/V', 'polyphen': 0.03, 'type': 'missense_variant', 'codons': 'gCg/gTg', 'allele': 'G/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 305, 'Parent': 'ENST00000288602', 'end': 305, 'id': 'rs547693139', 'seq_region_name': 'ENSP00000288602', 'sift': 0.39}, {'start': 587, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C', 'codons': 'Aaa/Gaa', 'polyphen': 0.998, 'residues': 'K/E', 'type': 'missense_variant', 'sift': 0.02, 'id': 'rs756706958', 'seq_region_name': 'ENSP00000288602', 'end': 587, 'Parent': 'ENST00000288602'}, {'sift': 0.02, 'Parent': 'ENST00000288602', 'id': 'rs1586186530', 'seq_region_name': 'ENSP00000288602', 'end': 359, 'allele': 'T/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'cAa/cGa', 'start': 359, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'polyphen': 0.824, 'type': 'missense_variant', 'residues': 'Q/R'}, {'Parent': 'ENST00000288602', 'end': 144, 'id': 'rs775527171', 'seq_region_name': 'ENSP00000288602', 'sift': 0.14, 'polyphen': 0.006, 'type': 'missense_variant', 'residues': 'V/M', 'codons': 'Gtg/Atg', 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 144}, {'type': 'missense_variant', 'polyphen': 0.001, 'residues': 'S/N', 'start': 419, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'aGt/aAt', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1033617391', 'end': 419, 'Parent': 'ENST00000288602', 'sift': 0.41}, {'Parent': 'ENST00000288602', 'id': 'rs1409797785', 'seq_region_name': 'ENSP00000288602', 'end': 440, 'sift': 0.31, 'residues': 'A/S', 'polyphen': 0.971, 'type': 'missense_variant', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/A', 'codons': 'Gct/Tct', 'start': 440, 'feature_type': 'transcript_variation', 'clinical_significance': []}, {'sift': 0, 'end': 632, 'seq_region_name': 'ENSP00000288602', 'id': 'rs121913363', 'Parent': 'ENST00000288602', 'clinical_significance': ['likely pathogenic'], 'feature_type': 'transcript_variation', 'start': 632, 'codons': 'Ata/Gta', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C', 'polyphen': 0.001, 'type': 'missense_variant', 'residues': 'I/V'}, {'residues': 'Q/E', 'polyphen': 0.995, 'type': 'missense_variant', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 356, 'codons': 'Caa/Gaa', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/C', 'end': 356, 'seq_region_name': 'ENSP00000288602', 'id': 'rs151308995', 'Parent': 'ENST00000288602', 'sift': 0.34}, {'start': 532, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A/T', 'codons': 'Cct/Tct/Act', 'residues': 'P/S/T', 'polyphen': 0.635, 'type': 'missense_variant', 'sift': 0.11, 'id': 'rs760614709', 'seq_region_name': 'ENSP00000288602', 'end': 532, 'Parent': 'ENST00000288602'}, {'sift': 0.28, 'seq_region_name': 'ENSP00000288602', 'id': 'rs748393685', 'end': 393, 'Parent': 'ENST00000288602', 'start': 393, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'codons': 'Gcc/Acc', 'residues': 'A/T', 'polyphen': 0.084, 'type': 'missense_variant'}, {'end': 525, 'seq_region_name': 'ENSP00000288602', 'id': 'rs180177036', 'Parent': 'ENST00000288602', 'sift': 0, 'residues': 'L/F/F', 'polyphen': 0.91, 'type': 'missense_variant', 'clinical_significance': ['uncertain significance', 'pathogenic'], 'feature_type': 'transcript_variation', 'start': 525, 'codons': 'ttG/ttT/ttC', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/A/G'}, {'sift': 0, 'Parent': 'ENST00000288602', 'id': 'rs368528867', 'seq_region_name': 'ENSP00000288602', 'end': 759, 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/A/T', 'codons': 'cGc/cTc/cAc', 'start': 759, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'type': 'missense_variant', 'polyphen': 0.007, 'residues': 'R/L/H'}, {'codons': 'gCa/gTa', 'allele': 'G/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 91, 'type': 'missense_variant', 'polyphen': 0.341, 'residues': 'A/V', 'sift': 0.18, 'Parent': 'ENST00000288602', 'end': 91, 'id': 'rs886041256', 'seq_region_name': 'ENSP00000288602'}, {'sift': 0.21, 'id': 'rs557241012', 'seq_region_name': 'ENSP00000288602', 'end': 146, 'Parent': 'ENST00000288602', 'start': 146, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': 0.0002, 'allele': 'C/T', 'codons': 'cGg/cAg', 'polyphen': 0.848, 'residues': 'R/Q', 'type': 'missense_variant'}, {'codons': 'gAg/gCg', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/G', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 24, 'polyphen': 0.223, 'type': 'missense_variant', 'residues': 'E/A', 'sift': 0.19, 'Parent': 'ENST00000288602', 'end': 24, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1586674580'}, {'polyphen': 0.105, 'type': 'missense_variant', 'residues': 'R/H', 'codons': 'cGt/cAt', 'minor_allele_frequency': None, 'allele': 'C/T', 'translation': 'ENSP00000288602', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 271, 'Parent': 'ENST00000288602', 'end': 271, 'seq_region_name': 'ENSP00000288602', 'id': 'rs730880413', 'sift': 0.58}, {'allele': 'G/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Ccc/Tcc', 'start': 345, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'residues': 'P/S', 'polyphen': 0.043, 'type': 'missense_variant', 'sift': 0.22, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs776453584', 'end': 345}, {'Parent': 'ENST00000288602', 'id': 'rs876661018', 'seq_region_name': 'ENSP00000288602', 'end': 87, 'sift': 0.05, 'polyphen': 0.511, 'residues': 'S/G', 'type': 'missense_variant', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'codons': 'Agc/Ggc', 'start': 87, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance']}, {'Parent': 'ENST00000288602', 'end': 352, 'id': 'rs201882896', 'seq_region_name': 'ENSP00000288602', 'sift': 0.23, 'type': 'missense_variant', 'polyphen': 0.999, 'residues': 'D/G', 'codons': 'gAt/gGt', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 352}, {'end': 505, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1417108573', 'Parent': 'ENST00000288602', 'sift': 0.18, 'polyphen': 0.127, 'type': 'missense_variant', 'residues': 'S/T', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 505, 'codons': 'Tct/Act', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/T'}, {'codons': 'gGc/gAc', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 558, 'polyphen': 1, 'residues': 'G/D', 'type': 'missense_variant', 'sift': 0, 'Parent': 'ENST00000288602', 'end': 558, 'id': 'rs1478587652', 'seq_region_name': 'ENSP00000288602'}, {'residues': 'E/D/E', 'polyphen': 0.007, 'type': 'missense_variant', 'codons': 'gaG/gaT/gaA', 'translation': 'ENSP00000288602', 'minor_allele_frequency': 0.009185, 'allele': 'C/A/T', 'clinical_significance': ['benign', 'likely benign'], 'feature_type': 'transcript_variation', 'start': 26, 'Parent': 'ENST00000288602', 'end': 26, 'id': 'rs371877084', 'seq_region_name': 'ENSP00000288602', 'sift': 0.66}, {'sift': 0, 'Parent': 'ENST00000288602', 'id': 'rs1206928359', 'seq_region_name': 'ENSP00000288602', 'end': 96, 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Gaa/Aaa', 'start': 96, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'polyphen': 0.998, 'residues': 'E/K', 'type': 'missense_variant'}, {'sift': 0.01, 'Parent': 'ENST00000288602', 'end': 252, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1325951163', 'codons': 'cGa/cCa/cAa', 'allele': 'C/G/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'start': 252, 'polyphen': 0.811, 'type': 'missense_variant', 'residues': 'R/P/Q'}, {'polyphen': 0.921, 'type': 'missense_variant', 'residues': 'S/Y', 'start': 122, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/T', 'codons': 'tCt/tAt', 'id': 'rs397507462', 'seq_region_name': 'ENSP00000288602', 'end': 122, 'Parent': 'ENST00000288602', 'sift': 0.08}, {'sift': 0, 'Parent': 'ENST00000288602', 'end': 646, 'id': 'rs1488384581', 'seq_region_name': 'ENSP00000288602', 'codons': 'gGg/gCg/gAg', 'minor_allele_frequency': None, 'allele': 'C/G/T', 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 646, 'type': 'missense_variant', 'polyphen': 0.999, 'residues': 'G/A/E'}, {'sift': 0.51, 'Parent': 'ENST00000288602', 'id': 'rs397507457', 'seq_region_name': 'ENSP00000288602', 'end': 26, 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'codons': 'Gag/Aag', 'start': 26, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely benign'], 'residues': 'E/K', 'polyphen': 0.135, 'type': 'missense_variant'}, {'codons': 'gAa/gTa/gGa/gCa', 'minor_allele_frequency': None, 'allele': 'T/A/C/G', 'translation': 'ENSP00000288602', 'clinical_significance': ['not provided', 'likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'start': 541, 'type': 'missense_variant', 'polyphen': 0.982, 'residues': 'E/V/G/A', 'sift': 0, 'Parent': 'ENST00000288602', 'end': 541, 'seq_region_name': 'ENSP00000288602', 'id': 'rs180177039'}, {'sift': 0.16, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1202874043', 'end': 204, 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/G', 'codons': 'gaG/gaC', 'start': 204, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'residues': 'E/D', 'polyphen': 0.364, 'type': 'missense_variant'}, {'sift': 0.18, 'Parent': 'ENST00000288602', 'id': 'rs147732750', 'seq_region_name': 'ENSP00000288602', 'end': 382, 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'A/T', 'codons': 'Ttg/Atg', 'start': 382, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'residues': 'L/M', 'polyphen': 0.813, 'type': 'missense_variant'}, {'type': 'missense_variant', 'polyphen': 0.868, 'residues': 'T/P', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/G', 'codons': 'Acc/Ccc', 'start': 244, 'clinical_significance': ['pathogenic'], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs397507465', 'end': 244, 'sift': 0.02}, {'sift': 0.01, 'id': 'rs397507470', 'seq_region_name': 'ENSP00000288602', 'end': 262, 'Parent': 'ENST00000288602', 'start': 262, 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'allele': 'G/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Caa/Aaa', 'polyphen': 0.695, 'type': 'missense_variant', 'residues': 'Q/K'}, {'start': 766, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'minor_allele_frequency': None, 'allele': 'C/T', 'translation': 'ENSP00000288602', 'codons': 'cGc/cAc', 'residues': 'R/H', 'polyphen': 0.999, 'type': 'missense_variant', 'sift': 0, 'seq_region_name': 'ENSP00000288602', 'id': 'rs777533608', 'end': 766, 'Parent': 'ENST00000288602'}, {'minor_allele_frequency': None, 'allele': 'C/A', 'translation': 'ENSP00000288602', 'codons': 'cGa/cTa', 'start': 362, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'polyphen': 0.999, 'type': 'missense_variant', 'residues': 'R/L', 'sift': 0.01, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1469574556', 'end': 362}, {'polyphen': 0.96, 'type': 'missense_variant', 'residues': 'A/P', 'start': 246, 'clinical_significance': ['pathogenic'], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/G', 'codons': 'Gca/Cca', 'seq_region_name': 'ENSP00000288602', 'id': 'rs180177034', 'end': 246, 'Parent': 'ENST00000288602', 'sift': 0}, {'sift': 0.25, 'id': 'rs766692331', 'seq_region_name': 'ENSP00000288602', 'end': 733, 'Parent': 'ENST00000288602', 'start': 733, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/G', 'codons': 'aTg/aCg', 'polyphen': 0.013, 'residues': 'M/T', 'type': 'missense_variant'}, {'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 622, 'codons': 'aTa/aCa', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'A/G', 'polyphen': 0.802, 'residues': 'I/T', 'type': 'missense_variant', 'sift': 0, 'end': 622, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1296743862', 'Parent': 'ENST00000288602'}, {'sift': 0.23, 'end': 331, 'id': 'rs372915332', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 331, 'codons': 'cTc/cCc', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/G', 'type': 'missense_variant', 'polyphen': 0.007, 'residues': 'L/P'}, {'end': 308, 'id': 'rs781001327', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.49, 'polyphen': 0.005, 'residues': 'A/T', 'type': 'missense_variant', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 308, 'codons': 'Gca/Aca', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T'}, {'sift': 0.03, 'Parent': 'ENST00000288602', 'end': 13, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1563042573', 'codons': 'Gag/Aag', 'minor_allele_frequency': None, 'allele': 'C/T', 'translation': 'ENSP00000288602', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 13, 'residues': 'E/K', 'polyphen': 0.138, 'type': 'missense_variant'}, {'sift': 0.02, 'end': 641, 'id': 'rs121913365', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic'], 'start': 641, 'codons': 'aaA/aaT/aaC', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/A/G', 'type': 'missense_variant', 'polyphen': 0.077, 'residues': 'K/N/N'}, {'sift': 0, 'Parent': 'ENST00000288602', 'end': 521, 'seq_region_name': 'ENSP00000288602', 'id': 'rs397516892', 'codons': 'gCa/gAa', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/T', 'clinical_significance': ['likely pathogenic'], 'feature_type': 'transcript_variation', 'start': 521, 'polyphen': 1, 'residues': 'A/E', 'type': 'missense_variant'}, {'end': 782, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1057520665', 'Parent': 'ENST00000288602', 'sift': 0, 'polyphen': 0.76, 'residues': 'D/N', 'type': 'missense_variant', 'clinical_significance': ['likely benign'], 'feature_type': 'transcript_variation', 'start': 782, 'codons': 'Gat/Aat', 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602'}, {'id': 'rs893650301', 'seq_region_name': 'ENSP00000288602', 'end': 721, 'Parent': 'ENST00000288602', 'sift': 0.16, 'type': 'missense_variant', 'polyphen': 0.184, 'residues': 'V/I', 'start': 721, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'codons': 'Gta/Ata'}, {'polyphen': 0.001, 'type': 'missense_variant', 'residues': 'A/T', 'codons': 'Gcg/Acg', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 305, 'Parent': 'ENST00000288602', 'end': 305, 'seq_region_name': 'ENSP00000288602', 'id': 'rs964235659', 'sift': 0.5}, {'residues': 'I/V', 'polyphen': 0.992, 'type': 'missense_variant', 'start': 77, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C', 'codons': 'Ata/Gta', 'seq_region_name': 'ENSP00000288602', 'id': 'rs764408896', 'end': 77, 'Parent': 'ENST00000288602', 'sift': 0.02}, {'end': 761, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1253996056', 'Parent': 'ENST00000288602', 'sift': 0, 'residues': 'L/F', 'polyphen': 0.934, 'type': 'missense_variant', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 761, 'codons': 'ttG/ttT', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/A'}, {'Parent': 'ENST00000288602', 'end': 386, 'id': 'rs1448515842', 'seq_region_name': 'ENSP00000288602', 'sift': 0.16, 'polyphen': 0.789, 'type': 'missense_variant', 'residues': 'Q/H', 'codons': 'caA/caC', 'allele': 'T/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 386}, {'type': 'missense_variant', 'polyphen': 0.347, 'residues': 'Q/R', 'minor_allele_frequency': None, 'allele': 'T/C', 'translation': 'ENSP00000288602', 'codons': 'cAa/cGa', 'start': 501, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1554399925', 'end': 501, 'sift': 0.35}, {'sift': 0, 'end': 276, 'id': 'rs1057518105', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic'], 'start': 276, 'codons': 'Gtt/Ctt', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/G', 'polyphen': 0.597, 'type': 'missense_variant', 'residues': 'V/L'}, {'residues': 'E/D/E', 'polyphen': 0.991, 'type': 'missense_variant', 'codons': 'gaG/gaC/gaA', 'allele': 'C/G/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 573, 'Parent': 'ENST00000288602', 'end': 573, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1357040745', 'sift': 0.06}, {'polyphen': 0.012, 'type': 'missense_variant', 'residues': 'T/S', 'codons': 'Act/Tct', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/A', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 310, 'Parent': 'ENST00000288602', 'end': 310, 'id': 'rs757243267', 'seq_region_name': 'ENSP00000288602', 'sift': 0.48}, {'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs794729219', 'end': 635, 'sift': 0, 'type': 'missense_variant', 'polyphen': 0.11, 'residues': 'F/L', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'A/G', 'codons': 'Ttt/Ctt', 'start': 635, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic']}, {'sift': 0, 'id': 'rs752322193', 'seq_region_name': 'ENSP00000288602', 'end': 699, 'Parent': 'ENST00000288602', 'start': 699, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/A', 'codons': 'Atc/Ttc', 'type': 'missense_variant', 'polyphen': 0.442, 'residues': 'I/F'}, {'codons': 'Tcg/Ccg', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/G', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 323, 'polyphen': 0.003, 'type': 'missense_variant', 'residues': 'S/P', 'sift': 0.27, 'Parent': 'ENST00000288602', 'end': 323, 'seq_region_name': 'ENSP00000288602', 'id': 'rs730880414'}, {'sift': 0.04, 'end': 283, 'id': 'rs756120029', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 283, 'codons': 'tAt/tGt', 'allele': 'T/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'residues': 'Y/C', 'polyphen': 0.882, 'type': 'missense_variant'}, {'sift': 0.05, 'Parent': 'ENST00000288602', 'end': 437, 'seq_region_name': 'ENSP00000288602', 'id': 'rs760702929', 'codons': 'Ggt/Tgt', 'minor_allele_frequency': None, 'allele': 'C/A', 'translation': 'ENSP00000288602', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 437, 'residues': 'G/C', 'polyphen': 0.998, 'type': 'missense_variant'}, {'Parent': 'ENST00000288602', 'end': 343, 'id': 'rs752186820', 'seq_region_name': 'ENSP00000288602', 'sift': 0.05, 'polyphen': 1, 'residues': 'P/L', 'type': 'missense_variant', 'codons': 'cCa/cTa', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 343}, {'minor_allele_frequency': None, 'allele': 'G/T', 'translation': 'ENSP00000288602', 'codons': 'gCa/gAa', 'start': 529, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'type': 'missense_variant', 'polyphen': 0.495, 'residues': 'A/E', 'sift': 0.71, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs905485545', 'end': 529}, {'sift': 0.02, 'seq_region_name': 'ENSP00000288602', 'id': 'rs397507456', 'end': 22, 'Parent': 'ENST00000288602', 'start': 22, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance', 'benign'], 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Gac/Aac', 'residues': 'D/N', 'polyphen': 0.997, 'type': 'missense_variant'}, {'sift': 0, 'seq_region_name': 'ENSP00000288602', 'id': 'rs55715359', 'end': 732, 'Parent': 'ENST00000288602', 'start': 732, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/G', 'codons': 'tTa/tCa', 'residues': 'L/S', 'polyphen': 0.958, 'type': 'missense_variant'}, {'type': 'missense_variant', 'polyphen': 0.938, 'residues': 'V/L/I', 'start': 157, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/G/T', 'codons': 'Gtt/Ctt/Att', 'id': 'rs1258111302', 'seq_region_name': 'ENSP00000288602', 'end': 157, 'Parent': 'ENST00000288602', 'sift': 0.29}, {'polyphen': 0.998, 'type': 'missense_variant', 'residues': 'K/R', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'codons': 'aAg/aGg', 'start': 513, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1345139126', 'end': 513, 'sift': 0.29}, {'sift': 0.07, 'id': 'rs757446039', 'seq_region_name': 'ENSP00000288602', 'end': 69, 'Parent': 'ENST00000288602', 'start': 69, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'codons': 'Ggt/Agt', 'type': 'missense_variant', 'polyphen': 0.933, 'residues': 'G/S'}, {'codons': 'Att/Gtt', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'clinical_significance': ['uncertain significance', 'likely benign'], 'feature_type': 'transcript_variation', 'start': 208, 'polyphen': 0.55, 'type': 'missense_variant', 'residues': 'I/V', 'sift': 0.18, 'Parent': 'ENST00000288602', 'end': 208, 'seq_region_name': 'ENSP00000288602', 'id': 'rs727504571'}, {'sift': 0.65, 'Parent': 'ENST00000288602', 'end': 338, 'seq_region_name': 'ENSP00000288602', 'id': 'rs756332987', 'codons': 'Aaa/Gaa', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 338, 'type': 'missense_variant', 'polyphen': 0.64, 'residues': 'K/E'}, {'type': 'missense_variant', 'polyphen': 0.005, 'residues': 'Q/R', 'start': 139, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'codons': 'cAa/cGa', 'seq_region_name': 'ENSP00000288602', 'id': 'rs774083294', 'end': 139, 'Parent': 'ENST00000288602', 'sift': 0.34}, {'sift': 0, 'Parent': 'ENST00000288602', 'end': 523, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1586126581', 'codons': 'aAa/aCa', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/G', 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic'], 'start': 523, 'polyphen': 1, 'type': 'missense_variant', 'residues': 'K/T'}, {'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 140, 'codons': 'aaT/aaG', 'allele': 'A/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'polyphen': 0.127, 'residues': 'N/K', 'type': 'missense_variant', 'sift': 0.8, 'end': 140, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1416091381', 'Parent': 'ENST00000288602'}, {'sift': 0, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs121913349', 'end': 504, 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/G/T', 'codons': 'Gga/Cga/Aga', 'start': 504, 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'polyphen': 0.998, 'type': 'missense_variant', 'residues': 'G/R/R'}, {'type': 'missense_variant', 'polyphen': 0.253, 'residues': 'S/G', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 127, 'codons': 'Agc/Ggc', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C', 'end': 127, 'id': 'rs543425316', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.17}, {'type': 'missense_variant', 'polyphen': 0.001, 'residues': 'P/A', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 493, 'codons': 'Cct/Gct', 'allele': 'G/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'end': 493, 'seq_region_name': 'ENSP00000288602', 'id': 'rs754904100', 'Parent': 'ENST00000288602', 'sift': 0.17}, {'id': 'rs1064796897', 'seq_region_name': 'ENSP00000288602', 'end': 23, 'Parent': 'ENST00000288602', 'sift': 0.54, 'polyphen': 0.025, 'residues': 'M/I', 'type': 'missense_variant', 'start': 23, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'allele': 'C/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'atG/atT'}, {'residues': 'S/L', 'polyphen': 0.021, 'type': 'missense_variant', 'codons': 'tCg/tTg', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 37, 'Parent': 'ENST00000288602', 'end': 37, 'id': 'rs1390903353', 'seq_region_name': 'ENSP00000288602', 'sift': 0.65}, {'type': 'missense_variant', 'polyphen': 0.007, 'residues': 'G/S', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 390, 'codons': 'Ggt/Agt', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'end': 390, 'seq_region_name': 'ENSP00000288602', 'id': 'rs886041424', 'Parent': 'ENST00000288602', 'sift': 0.57}, {'sift': 0, 'Parent': 'ENST00000288602', 'end': 545, 'id': 'rs397507477', 'seq_region_name': 'ENSP00000288602', 'codons': 'Ctc/Ttc', 'minor_allele_frequency': None, 'allele': 'G/A', 'translation': 'ENSP00000288602', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'start': 545, 'polyphen': 0.769, 'residues': 'L/F', 'type': 'missense_variant'}, {'sift': 0.01, 'Parent': 'ENST00000288602', 'end': 804, 'id': 'rs139420557', 'seq_region_name': 'ENSP00000288602', 'codons': 'cCt/cTt', 'allele': 'G/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 804, 'polyphen': 0.007, 'type': 'missense_variant', 'residues': 'P/L'}, {'Parent': 'ENST00000288602', 'id': 'rs1220982313', 'seq_region_name': 'ENSP00000288602', 'end': 551, 'sift': 0.04, 'type': 'missense_variant', 'polyphen': 1, 'residues': 'V/M', 'minor_allele_frequency': None, 'allele': 'C/T', 'translation': 'ENSP00000288602', 'codons': 'Gtg/Atg', 'start': 551, 'feature_type': 'transcript_variation', 'clinical_significance': []}, {'end': 712, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1428696172', 'Parent': 'ENST00000288602', 'sift': 0, 'polyphen': 1, 'residues': 'G/V', 'type': 'missense_variant', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 712, 'codons': 'gGa/gTa', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/A'}, {'polyphen': 0.007, 'type': 'missense_variant', 'residues': 'G/S', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 32, 'codons': 'Ggc/Agc', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'end': 32, 'id': 'rs1325363555', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.15}, {'polyphen': 0.292, 'residues': 'Q/K', 'type': 'missense_variant', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 94, 'codons': 'Caa/Aaa', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/T', 'end': 94, 'id': 'rs1002421360', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.89}, {'minor_allele_frequency': None, 'allele': 'G/T', 'translation': 'ENSP00000288602', 'codons': 'Caa/Aaa', 'start': 501, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'type': 'missense_variant', 'polyphen': 0.031, 'residues': 'Q/K', 'sift': 0.84, 'Parent': 'ENST00000288602', 'id': 'rs1185996698', 'seq_region_name': 'ENSP00000288602', 'end': 501}, {'allele': 'T/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'gAt/gTt', 'start': 519, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'type': 'missense_variant', 'polyphen': 0.981, 'residues': 'D/V', 'sift': 0.01, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1372569284', 'end': 519}, {'residues': 'L/S', 'polyphen': 0.743, 'type': 'missense_variant', 'allele': 'A/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'tTa/tCa', 'start': 456, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs780915629', 'end': 456, 'sift': 0.67}, {'type': 'missense_variant', 'polyphen': 0.999, 'residues': 'Q/H/Q', 'start': 153, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'allele': 'T/A/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'caA/caT/caG', 'id': 'rs1476241098', 'seq_region_name': 'ENSP00000288602', 'end': 153, 'Parent': 'ENST00000288602', 'sift': 0.21}, {'sift': 0, 'Parent': 'ENST00000288602', 'end': 241, 'id': 'rs387906661', 'seq_region_name': 'ENSP00000288602', 'codons': 'Acg/Ccg', 'allele': 'T/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['not provided', 'likely pathogenic', 'pathogenic'], 'start': 241, 'residues': 'T/P', 'polyphen': 0.937, 'type': 'missense_variant'}, {'residues': 'V/D/E', 'polyphen': 0.98, 'type': 'missense_variant', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'start': 640, 'codons': 'gTG/gAT/gAA', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'CA/AT/TT', 'end': 640, 'id': 'rs121913377', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0}, {'residues': 'G/V/A/E', 'polyphen': 0.987, 'type': 'missense_variant', 'start': 506, 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/A/G/T', 'codons': 'gGa/gTa/gCa/gAa', 'seq_region_name': 'ENSP00000288602', 'id': 'rs121913351', 'end': 506, 'Parent': 'ENST00000288602', 'sift': 0}, {'Parent': 'ENST00000288602', 'end': 19, 'id': 'rs745476335', 'seq_region_name': 'ENSP00000288602', 'sift': 0.99, 'type': 'missense_variant', 'polyphen': 0.001, 'residues': 'F/L', 'codons': 'Ttc/Ctc', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/G', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 19}, {'sift': 0.05, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1172916141', 'end': 499, 'minor_allele_frequency': None, 'allele': 'A/G', 'translation': 'ENSP00000288602', 'codons': 'gTg/gCg', 'start': 499, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'residues': 'V/A', 'polyphen': 0.003, 'type': 'missense_variant'}, {'sift': 0, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs397507484', 'end': 641, 'minor_allele_frequency': None, 'allele': 'T/A/C/G', 'translation': 'ENSP00000288602', 'codons': 'aAa/aTa/aGa/aCa', 'start': 641, 'clinical_significance': ['not provided', 'likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'residues': 'K/I/R/T', 'polyphen': 0.724, 'type': 'missense_variant'}, {'id': 'rs1239045029', 'seq_region_name': 'ENSP00000288602', 'end': 435, 'Parent': 'ENST00000288602', 'sift': 0.34, 'residues': 'T/N', 'polyphen': 0.697, 'type': 'missense_variant', 'start': 435, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'minor_allele_frequency': None, 'allele': 'G/T', 'translation': 'ENSP00000288602', 'codons': 'aCc/aAc'}, {'type': 'missense_variant', 'polyphen': 0.885, 'residues': 'A/D', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 789, 'codons': 'gCt/gAt', 'minor_allele_frequency': None, 'allele': 'G/T', 'translation': 'ENSP00000288602', 'end': 789, 'id': 'rs200490285', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.01}, {'sift': 0.02, 'Parent': 'ENST00000288602', 'end': 637, 'seq_region_name': 'ENSP00000288602', 'id': 'rs121913366', 'codons': 'cTa/cGa/cAa', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/C/T', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'start': 637, 'polyphen': 0.156, 'type': 'missense_variant', 'residues': 'L/R/Q'}, {'polyphen': 0.363, 'residues': 'S/F', 'type': 'missense_variant', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'codons': 'tCt/tTt', 'start': 416, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'Parent': 'ENST00000288602', 'id': 'rs974051255', 'seq_region_name': 'ENSP00000288602', 'end': 416, 'sift': 0.22}, {'end': 378, 'id': 'rs1278451495', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.02, 'residues': 'N/S', 'polyphen': 0.769, 'type': 'missense_variant', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 378, 'codons': 'aAt/aGt', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C'}, {'polyphen': 0.127, 'residues': 'G/D', 'type': 'missense_variant', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'codons': 'gGc/gAc', 'start': 32, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'Parent': 'ENST00000288602', 'id': 'rs730880417', 'seq_region_name': 'ENSP00000288602', 'end': 32, 'sift': 0.09}, {'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 20, 'codons': 'aAc/aGc', 'minor_allele_frequency': None, 'allele': 'T/C', 'translation': 'ENSP00000288602', 'type': 'missense_variant', 'polyphen': 0.992, 'residues': 'N/S', 'sift': 0, 'end': 20, 'id': 'rs781085650', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602'}, {'residues': 'G/E', 'polyphen': 0.475, 'type': 'missense_variant', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 203, 'codons': 'gGa/gAa', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'end': 203, 'seq_region_name': 'ENSP00000288602', 'id': 'rs777491533', 'Parent': 'ENST00000288602', 'sift': 0.04}, {'residues': 'G/C', 'polyphen': 1, 'type': 'missense_variant', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/A', 'codons': 'Ggt/Tgt', 'start': 6, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1209990009', 'end': 6, 'sift': 0.01}, {'residues': 'D/N', 'polyphen': 0.999, 'type': 'missense_variant', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 485, 'codons': 'Gac/Aac', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'end': 485, 'seq_region_name': 'ENSP00000288602', 'id': 'rs752429313', 'Parent': 'ENST00000288602', 'sift': 0.02}, {'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 93, 'codons': 'Caa/Aaa', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/T', 'residues': 'Q/K', 'polyphen': 0.992, 'type': 'missense_variant', 'sift': 0.06, 'end': 93, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1246425730', 'Parent': 'ENST00000288602'}, {'type': 'missense_variant', 'polyphen': 0.087, 'residues': 'T/I', 'start': 310, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'allele': 'G/A', 'translation': 'ENSP00000288602', 'codons': 'aCt/aTt', 'seq_region_name': 'ENSP00000288602', 'id': 'rs757803696', 'end': 310, 'Parent': 'ENST00000288602', 'sift': 0.39}, {'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 647, 'codons': 'tCc/tGc', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/C', 'polyphen': 0.011, 'type': 'missense_variant', 'residues': 'S/C', 'sift': 0.02, 'end': 647, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1255101216', 'Parent': 'ENST00000288602'}, {'sift': 0.21, 'Parent': 'ENST00000288602', 'id': 'rs1468404745', 'seq_region_name': 'ENSP00000288602', 'end': 336, 'minor_allele_frequency': None, 'allele': 'G/A/C', 'translation': 'ENSP00000288602', 'codons': 'Cct/Tct/Gct', 'start': 336, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'type': 'missense_variant', 'polyphen': 0.936, 'residues': 'P/S/A'}, {'codons': 'Tca/Gca', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/C', 'feature_type': 'transcript_variation', 'clinical_significance': ['pathogenic'], 'start': 507, 'polyphen': 0.184, 'residues': 'S/A', 'type': 'missense_variant', 'sift': 0, 'Parent': 'ENST00000288602', 'end': 507, 'id': 'rs869025606', 'seq_region_name': 'ENSP00000288602'}, {'start': 173, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'codons': 'tGt/tAt', 'type': 'missense_variant', 'polyphen': 0.005, 'residues': 'C/Y', 'sift': 0.2, 'id': 'rs1411975481', 'seq_region_name': 'ENSP00000288602', 'end': 173, 'Parent': 'ENST00000288602'}, {'polyphen': 0.994, 'type': 'missense_variant', 'residues': 'P/L', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'codons': 'cCt/cTt', 'start': 170, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'id': 'rs1409723450', 'seq_region_name': 'ENSP00000288602', 'end': 170, 'sift': 0.35}, {'sift': 0.01, 'end': 660, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1296245849', 'Parent': 'ENST00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 660, 'codons': 'Atg/Ttg/Gtg', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/A/C', 'polyphen': 0.003, 'type': 'missense_variant', 'residues': 'M/L/V'}, {'sift': 0, 'Parent': 'ENST00000288602', 'end': 508, 'id': 'rs397507473', 'seq_region_name': 'ENSP00000288602', 'codons': 'tTt/tCt', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'A/G', 'clinical_significance': ['pathogenic'], 'feature_type': 'transcript_variation', 'start': 508, 'polyphen': 0.702, 'type': 'missense_variant', 'residues': 'F/S'}, {'sift': 0, 'end': 241, 'seq_region_name': 'ENSP00000288602', 'id': 'rs387906660', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'start': 241, 'codons': 'aCg/aTg/aGg/aAg', 'minor_allele_frequency': None, 'allele': 'G/A/C/T', 'translation': 'ENSP00000288602', 'type': 'missense_variant', 'polyphen': 0.888, 'residues': 'T/M/R/K'}, {'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs781529187', 'end': 652, 'sift': 0, 'polyphen': 0.003, 'residues': 'Q/K', 'type': 'missense_variant', 'minor_allele_frequency': None, 'allele': 'G/T', 'translation': 'ENSP00000288602', 'codons': 'Cag/Aag', 'start': 652, 'clinical_significance': [], 'feature_type': 'transcript_variation'}, {'id': 'rs1562957703', 'seq_region_name': 'ENSP00000288602', 'end': 468, 'Parent': 'ENST00000288602', 'sift': 0.2, 'type': 'missense_variant', 'polyphen': 0.713, 'residues': 'S/L', 'start': 468, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'allele': 'G/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'tCa/tTa'}, {'sift': 0.31, 'Parent': 'ENST00000288602', 'end': 354, 'id': 'rs1158980679', 'seq_region_name': 'ENSP00000288602', 'codons': 'cGa/cAa', 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 354, 'type': 'missense_variant', 'polyphen': 0.072, 'residues': 'R/Q'}, {'sift': 0.06, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs781402121', 'end': 377, 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Gtc/Atc', 'start': 377, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'type': 'missense_variant', 'polyphen': 0.998, 'residues': 'V/I'}, {'sift': 0, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs121913341', 'end': 635, 'allele': 'A/C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'ttT/ttG/ttA', 'start': 635, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'type': 'missense_variant', 'polyphen': 0.11, 'residues': 'F/L/L'}, {'polyphen': 0.168, 'type': 'missense_variant', 'residues': 'E/K', 'start': 24, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'codons': 'Gag/Aag', 'seq_region_name': 'ENSP00000288602', 'id': 'rs730880416', 'end': 24, 'Parent': 'ENST00000288602', 'sift': 0.19}, {'minor_allele_frequency': None, 'allele': 'C/G', 'translation': 'ENSP00000288602', 'codons': 'gGg/gCg', 'start': 798, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'polyphen': 0.895, 'residues': 'G/A', 'type': 'missense_variant', 'sift': 0.01, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1444388837', 'end': 798}, {'allele': 'C/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Ggg/Cgg', 'start': 646, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'residues': 'G/R', 'polyphen': 1, 'type': 'missense_variant', 'sift': 0, 'Parent': 'ENST00000288602', 'id': 'rs868021367', 'seq_region_name': 'ENSP00000288602', 'end': 646}, {'Parent': 'ENST00000288602', 'end': 33, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1458837905', 'sift': 0.32, 'residues': 'A/P/T', 'polyphen': 0.737, 'type': 'missense_variant', 'codons': 'Gcc/Ccc/Acc', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/G/T', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 33}, {'Parent': 'ENST00000288602', 'id': 'rs121913336', 'seq_region_name': 'ENSP00000288602', 'end': 627, 'sift': 0.04, 'residues': 'D/E/E', 'polyphen': 0.12, 'type': 'missense_variant', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/C/T', 'codons': 'gaC/gaG/gaA', 'start': 627, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic']}, {'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs886041264', 'end': 538, 'sift': 0, 'residues': 'F/Y', 'polyphen': 0.998, 'type': 'missense_variant', 'allele': 'A/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'tTc/tAc', 'start': 538, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic']}, {'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'codons': 'tCa/tTa', 'start': 316, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'residues': 'S/L', 'polyphen': 0.003, 'type': 'missense_variant', 'sift': 0.35, 'Parent': 'ENST00000288602', 'id': 'rs1390750160', 'seq_region_name': 'ENSP00000288602', 'end': 316}, {'end': 331, 'id': 'rs1181422421', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.55, 'type': 'missense_variant', 'polyphen': 0.522, 'residues': 'L/F', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 331, 'codons': 'Ctc/Ttc', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A'}, {'type': 'missense_variant', 'polyphen': 0.003, 'residues': 'G/D', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'codons': 'gGc/gAc', 'start': 428, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'Parent': 'ENST00000288602', 'id': 'rs1374785778', 'seq_region_name': 'ENSP00000288602', 'end': 428, 'sift': 0.63}, {'Parent': 'ENST00000288602', 'id': 'rs1586155795', 'seq_region_name': 'ENSP00000288602', 'end': 400, 'sift': 0.08, 'polyphen': 0.286, 'type': 'missense_variant', 'residues': 'R/L', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/A', 'codons': 'cGc/cTc', 'start': 400, 'feature_type': 'transcript_variation', 'clinical_significance': []}, {'seq_region_name': 'ENSP00000288602', 'id': 'rs1161707003', 'end': 40, 'Parent': 'ENST00000288602', 'sift': 0.4, 'polyphen': 0.127, 'type': 'missense_variant', 'residues': 'D/N', 'start': 40, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'codons': 'Gac/Aac'}, {'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 798, 'codons': 'Ggg/Agg', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'type': 'missense_variant', 'polyphen': 0.422, 'residues': 'G/R', 'sift': 0.05, 'end': 798, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1348001321', 'Parent': 'ENST00000288602'}, {'codons': 'aGa/aTa', 'allele': 'C/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['pathogenic'], 'start': 502, 'residues': 'R/I', 'polyphen': 0.761, 'type': 'missense_variant', 'sift': 0, 'Parent': 'ENST00000288602', 'end': 502, 'seq_region_name': 'ENSP00000288602', 'id': 'rs180177032'}, {'end': 759, 'seq_region_name': 'ENSP00000288602', 'id': 'rs397507487', 'Parent': 'ENST00000288602', 'sift': 0, 'polyphen': 0.869, 'type': 'missense_variant', 'residues': 'R/C', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 759, 'codons': 'Cgc/Tgc', 'minor_allele_frequency': None, 'allele': 'G/A', 'translation': 'ENSP00000288602'}, {'residues': 'G/E', 'polyphen': 0.967, 'type': 'missense_variant', 'clinical_significance': ['likely pathogenic'], 'feature_type': 'transcript_variation', 'start': 799, 'codons': 'gGa/gAa', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'end': 799, 'id': 'rs1585930903', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.03}, {'sift': 0.5, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs575016745', 'end': 473, 'translation': 'ENSP00000288602', 'minor_allele_frequency': 0.0002, 'allele': 'T/G', 'codons': 'gaA/gaC', 'start': 473, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'type': 'missense_variant', 'polyphen': 0.028, 'residues': 'E/D'}, {'type': 'missense_variant', 'polyphen': 0.948, 'residues': 'G/R/R', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/G/T', 'codons': 'Ggg/Cgg/Agg', 'start': 21, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance', 'not provided'], 'Parent': 'ENST00000288602', 'id': 'rs587778113', 'seq_region_name': 'ENSP00000288602', 'end': 21, 'sift': 0}, {'sift': 0.01, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1586237312', 'end': 215, 'Parent': 'ENST00000288602', 'start': 215, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic'], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'codons': 'tCc/tTc', 'polyphen': 0.936, 'residues': 'S/F', 'type': 'missense_variant'}, {'residues': 'T/I/R', 'polyphen': 0.305, 'type': 'missense_variant', 'codons': 'aCa/aTa/aGa', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A/C', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance', 'likely pathogenic', 'pathogenic'], 'start': 639, 'Parent': 'ENST00000288602', 'end': 639, 'seq_region_name': 'ENSP00000288602', 'id': 'rs121913375', 'sift': 0}, {'codons': 'aaA/aaT/aaC', 'allele': 'T/A/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'start': 539, 'type': 'missense_variant', 'polyphen': 0.783, 'residues': 'K/N/N', 'sift': 0, 'Parent': 'ENST00000288602', 'end': 539, 'seq_region_name': 'ENSP00000288602', 'id': 'rs397507476'}, {'start': 795, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'allele': 'G/C', 'translation': 'ENSP00000288602', 'codons': 'atC/atG', 'type': 'missense_variant', 'polyphen': 0.999, 'residues': 'I/M', 'sift': 0.04, 'seq_region_name': 'ENSP00000288602', 'id': 'rs763340034', 'end': 795, 'Parent': 'ENST00000288602'}, {'polyphen': 0.811, 'type': 'missense_variant', 'residues': 'H/Q/Q', 'codons': 'caC/caG/caA', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/C/T', 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'start': 614, 'Parent': 'ENST00000288602', 'end': 614, 'id': 'rs397507481', 'seq_region_name': 'ENSP00000288602', 'sift': 0}, {'Parent': 'ENST00000288602', 'id': 'rs121913370', 'seq_region_name': 'ENSP00000288602', 'end': 621, 'sift': 0.04, 'polyphen': 0.003, 'type': 'missense_variant', 'residues': 'N/S/T', 'minor_allele_frequency': None, 'allele': 'T/C/G', 'translation': 'ENSP00000288602', 'codons': 'aAt/aGt/aCt', 'start': 621, 'clinical_significance': ['likely pathogenic'], 'feature_type': 'transcript_variation'}, {'start': 31, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely benign'], 'minor_allele_frequency': None, 'allele': 'G/A/C', 'translation': 'ENSP00000288602', 'codons': 'gCc/gTc/gGc', 'polyphen': 0.138, 'type': 'missense_variant', 'residues': 'A/V/G', 'sift': 0.33, 'seq_region_name': 'ENSP00000288602', 'id': 'rs397516906', 'end': 31, 'Parent': 'ENST00000288602'}, {'start': 509, 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'allele': 'CC/AA/GA', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'GGa/TTa/TCa', 'polyphen': 0.998, 'type': 'missense_variant', 'residues': 'G/L/S', 'sift': 0, 'id': 'rs1057519720', 'seq_region_name': 'ENSP00000288602', 'end': 509, 'Parent': 'ENST00000288602'}, {'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'start': 247, 'codons': 'ttT/ttG', 'minor_allele_frequency': None, 'allele': 'A/C', 'translation': 'ENSP00000288602', 'type': 'missense_variant', 'polyphen': 0.251, 'residues': 'F/L', 'sift': 0, 'end': 247, 'id': 'rs397509343', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602'}, {'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 464, 'codons': 'cGa/cAa', 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'polyphen': 0.939, 'residues': 'R/Q', 'type': 'missense_variant', 'sift': 0.01, 'end': 464, 'id': 'rs756851654', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602'}, {'polyphen': 0.087, 'type': 'missense_variant', 'residues': 'T/I', 'start': 332, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely benign'], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'codons': 'aCc/aTc', 'id': 'rs1008080053', 'seq_region_name': 'ENSP00000288602', 'end': 332, 'Parent': 'ENST00000288602', 'sift': 0.26}, {'polyphen': 0.951, 'residues': 'P/S', 'type': 'missense_variant', 'codons': 'Cct/Tct', 'minor_allele_frequency': None, 'allele': 'G/A', 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 530, 'Parent': 'ENST00000288602', 'end': 530, 'seq_region_name': 'ENSP00000288602', 'id': 'rs375520366', 'sift': 0}, {'polyphen': 0.931, 'residues': 'L/S', 'type': 'missense_variant', 'codons': 'CTa/TCa', 'minor_allele_frequency': None, 'allele': 'AG/GA', 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['pathogenic'], 'start': 637, 'Parent': 'ENST00000288602', 'end': 637, 'seq_region_name': 'ENSP00000288602', 'id': 'rs121913368', 'sift': 0}, {'polyphen': 0.988, 'type': 'missense_variant', 'residues': 'R/K', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'codons': 'aGg/aAg', 'start': 702, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'Parent': 'ENST00000288602', 'id': 'rs1317472067', 'seq_region_name': 'ENSP00000288602', 'end': 702, 'sift': 0.55}, {'end': 525, 'id': 'rs397507475', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0, 'polyphen': 0.984, 'residues': 'L/S', 'type': 'missense_variant', 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'start': 525, 'codons': 'tTg/tCg', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'A/G'}, {'end': 443, 'seq_region_name': 'ENSP00000288602', 'id': 'rs749792302', 'Parent': 'ENST00000288602', 'sift': 0.2, 'type': 'missense_variant', 'polyphen': 1, 'residues': 'P/S/A/T', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance', 'likely benign'], 'start': 443, 'codons': 'Cct/Tct/Gct/Act', 'allele': 'G/A/C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602'}, {'sift': 0.02, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1015872387', 'end': 220, 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'codons': 'Gaa/Aaa', 'start': 220, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'residues': 'E/K', 'polyphen': 0.792, 'type': 'missense_variant'}, {'polyphen': 0.978, 'type': 'missense_variant', 'residues': 'G/E', 'start': 106, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'codons': 'gGa/gAa', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1322163182', 'end': 106, 'Parent': 'ENST00000288602', 'sift': 0.04}, {'allele': 'CT/AA', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'AGt/TTt', 'start': 645, 'clinical_significance': ['likely pathogenic'], 'feature_type': 'transcript_variation', 'polyphen': 0.062, 'residues': 'S/F', 'type': 'missense_variant', 'sift': 0, 'Parent': 'ENST00000288602', 'id': 'rs121913372', 'seq_region_name': 'ENSP00000288602', 'end': 645}, {'type': 'missense_variant', 'polyphen': 0.997, 'residues': 'D/E/E', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/C/T', 'codons': 'gaT/gaG/gaA', 'start': 678, 'clinical_significance': ['not provided', 'likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'id': 'rs180177042', 'seq_region_name': 'ENSP00000288602', 'end': 678, 'sift': 0}, {'sift': 0, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1257391548', 'end': 745, 'Parent': 'ENST00000288602', 'start': 745, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'codons': 'cCa/cTa', 'type': 'missense_variant', 'polyphen': 0.926, 'residues': 'P/L'}, {'id': 'rs397516884', 'seq_region_name': 'ENSP00000288602', 'end': 347, 'Parent': 'ENST00000288602', 'sift': 0.09, 'polyphen': 0.936, 'type': 'missense_variant', 'residues': 'R/P/Q', 'start': 347, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'minor_allele_frequency': None, 'allele': 'C/G/T', 'translation': 'ENSP00000288602', 'codons': 'cGa/cCa/cAa'}, {'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A/T', 'codons': 'Cgg/Tgg/Agg', 'start': 146, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'residues': 'R/W/R', 'polyphen': 0.973, 'type': 'missense_variant', 'sift': 0.11, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs769648920', 'end': 146}, {'sift': 0.51, 'end': 136, 'seq_region_name': 'ENSP00000288602', 'id': 'rs138086018', 'Parent': 'ENST00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 136, 'codons': 'Tca/Gca', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/C', 'type': 'missense_variant', 'polyphen': 0.003, 'residues': 'S/A'}, {'sift': 0.24, 'Parent': 'ENST00000288602', 'end': 9, 'id': 'rs1211436028', 'seq_region_name': 'ENSP00000288602', 'codons': 'gGt/gCt', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/G', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 9, 'residues': 'G/A', 'polyphen': 0.079, 'type': 'missense_variant'}, {'polyphen': 0.006, 'type': 'missense_variant', 'residues': 'F/L', 'allele': 'A/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'ttT/ttG', 'start': 623, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs397507482', 'end': 623, 'sift': 0.03}, {'clinical_significance': ['pathogenic'], 'feature_type': 'transcript_variation', 'start': 512, 'codons': 'tAc/tGc', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C', 'polyphen': 0.896, 'residues': 'Y/C', 'type': 'missense_variant', 'sift': 0, 'end': 512, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1057519719', 'Parent': 'ENST00000288602'}, {'polyphen': 0.917, 'residues': 'Q/H/Q', 'type': 'missense_variant', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 344, 'codons': 'caG/caC/caA', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/G/T', 'end': 344, 'id': 'rs56094800', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.17}, {'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 341, 'codons': 'cCa/cTa/cAa', 'minor_allele_frequency': None, 'allele': 'G/A/T', 'translation': 'ENSP00000288602', 'polyphen': 0.944, 'residues': 'P/L/Q', 'type': 'missense_variant', 'sift': 0.06, 'end': 341, 'id': 'rs863224722', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602'}, {'residues': 'E/D/E', 'polyphen': 0.138, 'type': 'missense_variant', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 13, 'codons': 'gaG/gaC/gaA', 'minor_allele_frequency': None, 'allele': 'C/G/T', 'translation': 'ENSP00000288602', 'end': 13, 'seq_region_name': 'ENSP00000288602', 'id': 'rs868441785', 'Parent': 'ENST00000288602', 'sift': 0.07}, {'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 593, 'codons': 'Ctt/Ttt', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/A', 'polyphen': 1, 'type': 'missense_variant', 'residues': 'L/F', 'sift': 0.01, 'end': 593, 'seq_region_name': 'ENSP00000288602', 'id': 'rs749083550', 'Parent': 'ENST00000288602'}, {'polyphen': 0, 'residues': 'A/G', 'type': 'missense_variant', 'codons': 'gCc/gGc', 'allele': 'G/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 27, 'Parent': 'ENST00000288602', 'end': 27, 'id': 'rs1206505128', 'seq_region_name': 'ENSP00000288602', 'sift': 0.31}, {'sift': None, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1265184298', 'end': 688, 'Parent': 'ENST00000288602', 'start': 688, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'allele': 'T/C/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'gaA/gaG/gaC', 'residues': 'E/E/D', 'polyphen': None, 'type': 'missense_variant'}, {'id': 'rs1585931446', 'seq_region_name': 'ENSP00000288602', 'end': 770, 'Parent': 'ENST00000288602', 'sift': 0, 'polyphen': 0.482, 'residues': 'E/D', 'type': 'missense_variant', 'start': 770, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'minor_allele_frequency': None, 'allele': 'T/A', 'translation': 'ENSP00000288602', 'codons': 'gaA/gaT'}, {'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1317540371', 'end': 349, 'sift': 0.15, 'residues': 'A/V', 'polyphen': 0.341, 'type': 'missense_variant', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'codons': 'gCa/gTa', 'start': 349, 'clinical_significance': [], 'feature_type': 'transcript_variation'}, {'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs759901913', 'end': 524, 'sift': 1, 'residues': 'M/I', 'polyphen': 0.01, 'type': 'missense_variant', 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'atG/atA', 'start': 524, 'feature_type': 'transcript_variation', 'clinical_significance': []}, {'sift': 0.04, 'Parent': 'ENST00000288602', 'id': 'rs878854675', 'seq_region_name': 'ENSP00000288602', 'end': 594, 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'A/G', 'codons': 'aTa/aCa', 'start': 594, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'residues': 'I/T', 'polyphen': 0.144, 'type': 'missense_variant'}, {'sift': 0.25, 'end': 178, 'id': 'rs746348396', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 178, 'codons': 'cGa/cTa/cAa', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/A/T', 'polyphen': 0.063, 'type': 'missense_variant', 'residues': 'R/L/Q'}, {'sift': 0.51, 'end': 173, 'id': 'rs749677118', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 173, 'codons': 'Tgt/Cgt', 'minor_allele_frequency': None, 'allele': 'A/G', 'translation': 'ENSP00000288602', 'polyphen': 0.243, 'residues': 'C/R', 'type': 'missense_variant'}, {'residues': 'F/S', 'polyphen': 0.646, 'type': 'missense_variant', 'start': 635, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic'], 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'A/G', 'codons': 'tTt/tCt', 'id': 'rs121913225', 'seq_region_name': 'ENSP00000288602', 'end': 635, 'Parent': 'ENST00000288602', 'sift': 0}, {'sift': 0, 'Parent': 'ENST00000288602', 'end': 265, 'id': 'rs397516905', 'seq_region_name': 'ENSP00000288602', 'codons': 'Ggt/Cgt', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/G', 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic'], 'start': 265, 'type': 'missense_variant', 'polyphen': 0.885, 'residues': 'G/R'}, {'end': 750, 'id': 'rs1585931858', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0, 'residues': 'I/F', 'polyphen': 0.548, 'type': 'missense_variant', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 750, 'codons': 'Att/Ttt', 'allele': 'T/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602'}, {'end': 133, 'id': 'rs1057517889', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.21, 'polyphen': 0.049, 'type': 'missense_variant', 'residues': 'S/L', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 133, 'codons': 'tCa/tTa', 'minor_allele_frequency': None, 'allele': 'G/A', 'translation': 'ENSP00000288602'}, {'sift': 0.8, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs756114784', 'end': 150, 'allele': 'T/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'aAg/aGg', 'start': 150, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'type': 'missense_variant', 'polyphen': 0.045, 'residues': 'K/R'}, {'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 44, 'codons': 'cCg/cTg', 'minor_allele_frequency': None, 'allele': 'G/A', 'translation': 'ENSP00000288602', 'polyphen': 0.396, 'residues': 'P/L', 'type': 'missense_variant', 'sift': 0.22, 'end': 44, 'seq_region_name': 'ENSP00000288602', 'id': 'rs794726917', 'Parent': 'ENST00000288602'}, {'codons': 'Gca/Aca', 'minor_allele_frequency': None, 'allele': 'C/T', 'translation': 'ENSP00000288602', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 320, 'polyphen': 0.003, 'residues': 'A/T', 'type': 'missense_variant', 'sift': 0.43, 'Parent': 'ENST00000288602', 'end': 320, 'id': 'rs1461617552', 'seq_region_name': 'ENSP00000288602'}, {'sift': 0, 'Parent': 'ENST00000288602', 'end': 510, 'id': 'rs397516891', 'seq_region_name': 'ENSP00000288602', 'codons': 'aCa/aGa', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/C', 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic'], 'start': 510, 'polyphen': 1, 'residues': 'T/R', 'type': 'missense_variant'}, {'start': 323, 'clinical_significance': ['uncertain significance', 'benign'], 'feature_type': 'transcript_variation', 'allele': 'G/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'tCg/tTg', 'residues': 'S/L', 'polyphen': 0.021, 'type': 'missense_variant', 'sift': 0.18, 'id': 'rs397516907', 'seq_region_name': 'ENSP00000288602', 'end': 323, 'Parent': 'ENST00000288602'}, {'polyphen': 0.413, 'residues': 'S/F/C', 'type': 'missense_variant', 'codons': 'tCt/tTt/tGt', 'minor_allele_frequency': None, 'allele': 'G/A/C', 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 102, 'Parent': 'ENST00000288602', 'end': 102, 'id': 'rs774142136', 'seq_region_name': 'ENSP00000288602', 'sift': 0.11}, {'sift': 0.46, 'Parent': 'ENST00000288602', 'end': 301, 'id': 'rs34776339', 'seq_region_name': 'ENSP00000288602', 'codons': 'Cca/Tca', 'minor_allele_frequency': None, 'allele': 'G/A', 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 301, 'residues': 'P/S', 'polyphen': 0.003, 'type': 'missense_variant'}, {'residues': 'E/K', 'polyphen': 0.267, 'type': 'missense_variant', 'codons': 'Gag/Aag', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 426, 'Parent': 'ENST00000288602', 'end': 426, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1329402179', 'sift': 0.52}, {'sift': 0.02, 'end': 507, 'id': 'rs867748453', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 507, 'codons': 'tCa/tTa', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/A', 'residues': 'S/L', 'polyphen': 0.547, 'type': 'missense_variant'}, {'polyphen': 0.394, 'residues': 'A/P', 'type': 'missense_variant', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/G', 'codons': 'Gct/Cct', 'start': 38, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'Parent': 'ENST00000288602', 'id': 'rs1011563467', 'seq_region_name': 'ENSP00000288602', 'end': 38, 'sift': 0.28}, {'Parent': 'ENST00000288602', 'id': 'rs397516894', 'seq_region_name': 'ENSP00000288602', 'end': 614, 'sift': 0, 'type': 'missense_variant', 'polyphen': 0.844, 'residues': 'H/Y/D', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A/C', 'codons': 'Cac/Tac/Gac', 'start': 614, 'feature_type': 'transcript_variation', 'clinical_significance': ['not provided', 'pathogenic']}, {'sift': 0.37, 'Parent': 'ENST00000288602', 'end': 777, 'id': 'rs778872326', 'seq_region_name': 'ENSP00000288602', 'codons': 'gGt/gAt', 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 777, 'polyphen': 0.997, 'residues': 'G/D', 'type': 'missense_variant'}, {'codons': 'Gaa/Aaa', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'start': 83, 'type': 'missense_variant', 'polyphen': 0.998, 'residues': 'E/K', 'sift': 0.01, 'Parent': 'ENST00000288602', 'end': 83, 'seq_region_name': 'ENSP00000288602', 'id': 'rs753354016'}, {'sift': 0.64, 'id': 'rs1554404461', 'seq_region_name': 'ENSP00000288602', 'end': 218, 'Parent': 'ENST00000288602', 'start': 218, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'allele': 'G/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'aCt/aGt', 'type': 'missense_variant', 'polyphen': 0.465, 'residues': 'T/S'}, {'codons': 'Gtt/Att', 'minor_allele_frequency': None, 'allele': 'C/T', 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 685, 'type': 'missense_variant', 'polyphen': 0.987, 'residues': 'V/I', 'sift': 0, 'Parent': 'ENST00000288602', 'end': 685, 'seq_region_name': 'ENSP00000288602', 'id': 'rs778397955'}, {'sift': 1, 'end': 145, 'id': 'rs1369170225', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 145, 'codons': 'Gca/Tca/Aca', 'minor_allele_frequency': None, 'allele': 'C/A/T', 'translation': 'ENSP00000288602', 'polyphen': 0, 'type': 'missense_variant', 'residues': 'A/S/T'}, {'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 321, 'codons': 'Ccc/Tcc', 'minor_allele_frequency': None, 'allele': 'G/A', 'translation': 'ENSP00000288602', 'residues': 'P/S', 'polyphen': 0.038, 'type': 'missense_variant', 'sift': 0.12, 'end': 321, 'seq_region_name': 'ENSP00000288602', 'id': 'rs760952736', 'Parent': 'ENST00000288602'}, {'id': 'rs1554389828', 'seq_region_name': 'ENSP00000288602', 'end': 749, 'Parent': 'ENST00000288602', 'sift': 0.01, 'polyphen': 0.981, 'type': 'missense_variant', 'residues': 'Q/K', 'start': 749, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/T', 'codons': 'Caa/Aaa'}, {'sift': 0.07, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1301589020', 'end': 345, 'Parent': 'ENST00000288602', 'start': 345, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'minor_allele_frequency': None, 'allele': 'G/C', 'translation': 'ENSP00000288602', 'codons': 'cCc/cGc', 'polyphen': 0.717, 'residues': 'P/R', 'type': 'missense_variant'}, {'start': 627, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic'], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/G', 'codons': 'gAc/gCc', 'residues': 'D/A', 'polyphen': 0.173, 'type': 'missense_variant', 'sift': 0.03, 'id': 'rs121913335', 'seq_region_name': 'ENSP00000288602', 'end': 627, 'Parent': 'ENST00000288602'}, {'start': 59, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T', 'codons': 'Gaa/Aaa', 'polyphen': 0.673, 'residues': 'E/K', 'type': 'missense_variant', 'sift': 0.18, 'seq_region_name': 'ENSP00000288602', 'id': 'rs371504803', 'end': 59, 'Parent': 'ENST00000288602'}, {'Parent': 'ENST00000288602', 'id': 'rs1586015221', 'seq_region_name': 'ENSP00000288602', 'end': 624, 'sift': 0, 'polyphen': 0.852, 'residues': 'L/F', 'type': 'missense_variant', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'codons': 'Ctt/Ttt', 'start': 624, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic']}, {'codons': 'Aca/Tca', 'allele': 'T/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 531, 'polyphen': 0.257, 'type': 'missense_variant', 'residues': 'T/S', 'sift': 0.1, 'Parent': 'ENST00000288602', 'end': 531, 'seq_region_name': 'ENSP00000288602', 'id': 'rs777038089'}, {'id': 'rs397516893', 'seq_region_name': 'ENSP00000288602', 'end': 527, 'Parent': 'ENST00000288602', 'sift': 0, 'polyphen': 0.783, 'residues': 'V/G', 'type': 'missense_variant', 'start': 527, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'minor_allele_frequency': None, 'allele': 'A/C', 'translation': 'ENSP00000288602', 'codons': 'gTg/gGg'}, {'type': 'missense_variant', 'polyphen': 0.898, 'residues': 'G/V/D', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'start': 636, 'codons': 'gGt/gTt/gAt', 'minor_allele_frequency': None, 'allele': 'C/A/T', 'translation': 'ENSP00000288602', 'end': 636, 'seq_region_name': 'ENSP00000288602', 'id': 'rs397507483', 'Parent': 'ENST00000288602', 'sift': 0}, {'sift': 0.01, 'seq_region_name': 'ENSP00000288602', 'id': 'rs746778122', 'end': 23, 'Parent': 'ENST00000288602', 'start': 23, 'clinical_significance': ['uncertain significance', 'benign'], 'feature_type': 'transcript_variation', 'allele': 'A/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'aTg/aAg', 'polyphen': 0.222, 'residues': 'M/K', 'type': 'missense_variant'}, {'residues': 'Q/Q/H', 'polyphen': None, 'type': 'missense_variant', 'start': 93, 'feature_type': 'transcript_variation', 'clinical_significance': ['likely benign'], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/C/G', 'codons': 'caA/caG/caC', 'seq_region_name': 'ENSP00000288602', 'id': 'rs150050723', 'end': 93, 'Parent': 'ENST00000288602', 'sift': None}, {'Parent': 'ENST00000288602', 'end': 616, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1057517915', 'sift': 0, 'polyphen': 0.743, 'type': 'missense_variant', 'residues': 'D/Y', 'codons': 'Gac/Tac', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/A', 'clinical_significance': ['likely pathogenic'], 'feature_type': 'transcript_variation', 'start': 616}, {'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1382327548', 'end': 315, 'sift': 0.39, 'residues': 'G/R', 'polyphen': 0.007, 'type': 'missense_variant', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/T', 'codons': 'Gga/Aga', 'start': 315, 'feature_type': 'transcript_variation', 'clinical_significance': []}, {'end': 522, 'id': 'rs765715150', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0, 'type': 'missense_variant', 'polyphen': 0.98, 'residues': 'V/M', 'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 522, 'codons': 'Gtg/Atg', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/T'}, {'residues': 'H/Y', 'polyphen': 0.423, 'type': 'missense_variant', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 806, 'codons': 'Cac/Tac', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'end': 806, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1317051737', 'Parent': 'ENST00000288602', 'sift': 0.66}, {'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 85, 'codons': 'tAc/tGc', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/C', 'residues': 'Y/C', 'polyphen': 0.967, 'type': 'missense_variant', 'sift': 0.01, 'end': 85, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1131691387', 'Parent': 'ENST00000288602'}, {'sift': 0.26, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1586209574', 'end': 317, 'Parent': 'ENST00000288602', 'start': 317, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'A/G', 'codons': 'Tcc/Ccc', 'polyphen': 0.207, 'type': 'missense_variant', 'residues': 'S/P'}, {'codons': 'gaG/gaC/gaA', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/G/T', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance', 'not provided', 'likely benign'], 'start': 24, 'residues': 'E/D/E', 'polyphen': 0.009, 'type': 'missense_variant', 'sift': 0.47, 'Parent': 'ENST00000288602', 'end': 24, 'id': 'rs587778114', 'seq_region_name': 'ENSP00000288602'}, {'sift': 0.2, 'id': 'rs730880418', 'seq_region_name': 'ENSP00000288602', 'end': 334, 'Parent': 'ENST00000288602', 'start': 334, 'clinical_significance': ['uncertain significance'], 'feature_type': 'transcript_variation', 'allele': 'G/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'cCg/cTg', 'type': 'missense_variant', 'polyphen': 0.092, 'residues': 'P/L'}, {'sift': 0.2, 'end': 87, 'id': 'rs1033856250', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 87, 'codons': 'aGc/aAc', 'minor_allele_frequency': None, 'allele': 'C/T', 'translation': 'ENSP00000288602', 'polyphen': 0.073, 'type': 'missense_variant', 'residues': 'S/N'}, {'polyphen': 0.377, 'residues': 'P/L', 'type': 'missense_variant', 'start': 429, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'codons': 'cCa/cTa', 'seq_region_name': 'ENSP00000288602', 'id': 'rs1586155498', 'end': 429, 'Parent': 'ENST00000288602', 'sift': 0.13}, {'start': 3, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'allele': 'C/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Gcg/Acg', 'polyphen': 0.997, 'residues': 'A/T', 'type': 'missense_variant', 'sift': 0, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1248316144', 'end': 3, 'Parent': 'ENST00000288602'}, {'seq_region_name': 'ENSP00000288602', 'id': 'rs746074624', 'end': 662, 'Parent': 'ENST00000288602', 'sift': 0.07, 'residues': 'P/A', 'polyphen': 0.684, 'type': 'missense_variant', 'start': 662, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'allele': 'G/C', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Cca/Gca'}, {'polyphen': 0.003, 'residues': 'A/V', 'type': 'missense_variant', 'codons': 'gCt/gTt', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'G/A', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 38, 'Parent': 'ENST00000288602', 'end': 38, 'seq_region_name': 'ENSP00000288602', 'id': 'rs886042293', 'sift': 0.31}, {'residues': 'N/K', 'polyphen': 0.864, 'type': 'missense_variant', 'minor_allele_frequency': None, 'allele': 'A/T', 'translation': 'ENSP00000288602', 'codons': 'aaT/aaA', 'start': 378, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'id': 'rs889203931', 'seq_region_name': 'ENSP00000288602', 'end': 378, 'sift': 0.02}, {'type': 'missense_variant', 'polyphen': 1, 'residues': 'W/C/C', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/A/G', 'codons': 'tgG/tgT/tgC', 'start': 571, 'feature_type': 'transcript_variation', 'clinical_significance': ['pathogenic'], 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs606231228', 'end': 571, 'sift': 0}, {'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'A/C', 'codons': 'gaT/gaG', 'start': 605, 'clinical_significance': ['likely pathogenic'], 'feature_type': 'transcript_variation', 'residues': 'D/E', 'polyphen': 0.04, 'type': 'missense_variant', 'sift': 0.04, 'Parent': 'ENST00000288602', 'id': 'rs397507480', 'seq_region_name': 'ENSP00000288602', 'end': 605}, {'seq_region_name': 'ENSP00000288602', 'id': 'rs397507471', 'end': 374, 'Parent': 'ENST00000288602', 'sift': 0.1, 'residues': 'I/L/V', 'polyphen': 0.486, 'type': 'missense_variant', 'start': 374, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/A/C', 'codons': 'Ata/Tta/Gta'}, {'start': 276, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'A/G', 'codons': 'gTt/gCt', 'residues': 'V/A', 'polyphen': 0.511, 'type': 'missense_variant', 'sift': 0, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1409250783', 'end': 276, 'Parent': 'ENST00000288602'}, {'allele': 'G/A/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'gCc/gTc/gAc', 'start': 752, 'clinical_significance': ['uncertain significance', 'likely pathogenic'], 'feature_type': 'transcript_variation', 'polyphen': 0.436, 'type': 'missense_variant', 'residues': 'A/V/D', 'sift': 0.17, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs727502904', 'end': 752}, {'polyphen': 0.983, 'residues': 'M/I', 'type': 'missense_variant', 'start': 690, 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'C/G', 'codons': 'atG/atC', 'seq_region_name': 'ENSP00000288602', 'id': 'rs879255380', 'end': 690, 'Parent': 'ENST00000288602', 'sift': 0.02}, {'id': 'rs1026592386', 'seq_region_name': 'ENSP00000288602', 'end': 415, 'Parent': 'ENST00000288602', 'sift': 0.15, 'residues': 'H/L', 'polyphen': 0.039, 'type': 'missense_variant', 'start': 415, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'T/A', 'codons': 'cAt/cTt'}, {'polyphen': 0.615, 'type': 'missense_variant', 'residues': 'F/V/L', 'allele': 'A/C/G', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Ttt/Gtt/Ctt', 'start': 247, 'clinical_significance': ['uncertain significance', 'likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'id': 'rs397516903', 'seq_region_name': 'ENSP00000288602', 'end': 247, 'sift': 0}, {'seq_region_name': 'ENSP00000288602', 'id': 'rs121913227', 'end': 640, 'Parent': 'ENST00000288602', 'sift': 0, 'residues': 'V/R/K', 'polyphen': 0.986, 'type': 'missense_variant', 'start': 640, 'feature_type': 'transcript_variation', 'clinical_significance': ['pathogenic', 'drug response'], 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'AC/CT/TT', 'codons': 'GTg/AGg/AAg'}, {'end': 386, 'id': 'rs1554401118', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0.07, 'type': 'missense_variant', 'polyphen': 0.444, 'residues': 'Q/R', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 386, 'codons': 'cAa/cGa', 'minor_allele_frequency': None, 'allele': 'T/C', 'translation': 'ENSP00000288602'}, {'sift': 0.02, 'end': 596, 'id': 'rs760524720', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance', 'likely benign'], 'start': 596, 'codons': 'aTt/aCt', 'minor_allele_frequency': None, 'allele': 'A/G', 'translation': 'ENSP00000288602', 'type': 'missense_variant', 'polyphen': 0.868, 'residues': 'I/T'}, {'feature_type': 'transcript_variation', 'clinical_significance': ['uncertain significance'], 'start': 344, 'codons': 'cAg/cTg/cCg', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/A/G', 'type': 'missense_variant', 'polyphen': 0.03, 'residues': 'Q/L/P', 'sift': 0.28, 'end': 344, 'id': 'rs758920663', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602'}, {'type': 'missense_variant', 'polyphen': 0.007, 'residues': 'V/L/L/M', 'allele': 'C/A/G/T', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'Gtg/Ttg/Ctg/Atg', 'start': 640, 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'feature_type': 'transcript_variation', 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs121913378', 'end': 640, 'sift': 0.02}, {'sift': 0, 'Parent': 'ENST00000288602', 'seq_region_name': 'ENSP00000288602', 'id': 'rs758134174', 'end': 772, 'allele': 'G/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'codons': 'tCc/tTc', 'start': 772, 'feature_type': 'transcript_variation', 'clinical_significance': [], 'type': 'missense_variant', 'polyphen': 0.999, 'residues': 'S/F'}, {'sift': 0, 'seq_region_name': 'ENSP00000288602', 'id': 'rs764565088', 'end': 795, 'Parent': 'ENST00000288602', 'start': 795, 'clinical_significance': [], 'feature_type': 'transcript_variation', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'A/G', 'codons': 'aTc/aCc', 'polyphen': 0.999, 'residues': 'I/T', 'type': 'missense_variant'}, {'type': 'missense_variant', 'polyphen': 0.999, 'residues': 'G/V/A/E', 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic', 'pathogenic'], 'start': 509, 'codons': 'gGa/gTa/gCa/gAa', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'C/A/G/T', 'end': 509, 'id': 'rs121913355', 'seq_region_name': 'ENSP00000288602', 'Parent': 'ENST00000288602', 'sift': 0}, {'Parent': 'ENST00000288602', 'id': 'rs757016514', 'seq_region_name': 'ENSP00000288602', 'end': 709, 'sift': 0, 'polyphen': 0.69, 'type': 'missense_variant', 'residues': 'V/A', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'A/G', 'codons': 'gTg/gCg', 'start': 709, 'feature_type': 'transcript_variation', 'clinical_significance': []}, {'feature_type': 'transcript_variation', 'clinical_significance': [], 'start': 441, 'codons': 'Acc/Ccc', 'translation': 'ENSP00000288602', 'minor_allele_frequency': None, 'allele': 'T/G', 'type': 'missense_variant', 'polyphen': 0.994, 'residues': 'T/P', 'sift': 0.21, 'end': 441, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1586146757', 'Parent': 'ENST00000288602'}, {'sift': 0.01, 'Parent': 'ENST00000288602', 'end': 484, 'id': 'rs121913371', 'seq_region_name': 'ENSP00000288602', 'codons': 'Cgg/Tgg/Agg', 'minor_allele_frequency': None, 'allele': 'G/A/T', 'translation': 'ENSP00000288602', 'feature_type': 'transcript_variation', 'clinical_significance': ['likely pathogenic'], 'start': 484, 'polyphen': 0.112, 'type': 'missense_variant', 'residues': 'R/W/R'}, {'sift': 0.16, 'Parent': 'ENST00000288602', 'end': 462, 'seq_region_name': 'ENSP00000288602', 'id': 'rs1381121699', 'codons': 'Cct/Tct', 'minor_allele_frequency': None, 'translation': 'ENSP00000288602', 'allele': 'G/A', 'clinical_significance': [], 'feature_type': 'transcript_variation', 'start': 462, 'polyphen': 0.272, 'residues': 'P/S', 'type': 'missense_variant'}]\n"
     ]
    }
   ],
   "source": [
    "import requests, sys\n",
    "\n",
    "server = \"https://rest.ensembl.org\"\n",
    "ext = \"/overlap/translation/ENSP00000288602?feature=transcript_variation;type=missense_variant\"\n",
    "\n",
    "r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"})\n",
    "\n",
    "if not r.ok:\n",
    "  r.raise_for_status()\n",
    "  sys.exit()\n",
    "\n",
    "decoded = r.json()\n",
    "print(repr(decoded))\n"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 8/8 [00:24<00:00,  3.09s/it]\n"
     ]
    }
   ],
   "source": [
    "data_list = list()\n",
    "for ds in dataset_list:\n",
    "    if ds.endogenous:\n",
    "        data_list.append(ds)\n",
    "\n",
    "design_list = list()\n",
    "for ds in tqdm(data_list):\n",
    "    ds.load_data()\n",
    "    ds.set_sgrnas()\n",
    "    design_list.append(ds.get_designs())"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "outputs": [
    {
     "data": {
      "text/plain": "2998"
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "design_df = (pd.concat(design_list)\n",
    "             .drop_duplicates())\n",
    "transcripts = (design_df['Target Transcript']\n",
    "               .str.split('.', expand=True)[0]\n",
    "               .unique())\n",
    "len(transcripts)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 60/60 [05:12<00:00,  5.21s/it]\n"
     ]
    }
   ],
   "source": [
    "transcipt_sequence_list = []\n",
    "for chunk_transcripts in tqdm(np.array_split(transcripts, np.ceil(len(transcripts)/50))):\n",
    "    transcipt_sequence_list.append(post_sequence_ids(list(chunk_transcripts)))"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "Missing genes appear primarily to be outdated transcript IDs"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%% md\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Missing: ENST00000611665, ENST00000622530, ENST00000368563, ENST00000377815, ENST00000618014, ENST00000424325, ENST00000650726, ENST00000572844, ENST00000344894, ENST00000355883, ENST00000648169, ENST00000449977\n"
     ]
    }
   ],
   "source": [
    "transcript_sequence_df = (pd.concat(transcipt_sequence_list)\n",
    "                          .reset_index(drop=True)\n",
    "                          .rename({'query': 'Transcript Base'}, axis=1))\n",
    "transcript_series = pd.Series(transcripts)\n",
    "missing_sequences = transcript_series[~transcript_series.isin(transcript_sequence_df['Transcript Base'])]\n",
    "print('Missing: ' + ', '.join(missing_sequences))\n"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "## Get Protein Domains\n",
    "\n",
    "Get all annotated protein domains based on ptrotein ID. Information about annotations can be found here: http://m.ensembl.org/Help/View?id=178"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%% md\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 2986/2986 [38:36<00:00,  1.29it/s] \n"
     ]
    }
   ],
   "source": [
    "translation_overlap_list = []\n",
    "for protein_id in tqdm(transcript_sequence_df['id'].unique()):\n",
    "    translation_overlap_list.append(get_translation_overlap(protein_id))"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "outputs": [
    {
     "data": {
      "text/plain": "sifts               18687\nSeg                  9515\nMobiDBLite           7753\nPfam                 5967\nPANTHER              5309\nGene3D               5049\nSuperFamily          3898\nSmart                3772\nProsite_profiles     3111\nPRINTS               2855\nCDD                  2321\nncoils               2178\nProsite_patterns     1308\nTMHMM                 539\nTIGRfam               429\nPIRSF                 423\nHAMAP                 285\nSignalP                56\nSFLD                   46\nName: type, dtype: int64"
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "translation_overlap_df = (pd.concat(translation_overlap_list)\n",
    "                          .reset_index(drop=True)\n",
    "                          .rename({'Parent': 'Transcript Base'}, axis=1))\n",
    "translation_overlap_df['type'].value_counts()"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "outputs": [
    {
     "data": {
      "text/plain": "seg            9515\nmobidb-lite    7753\nCoil           2178\nSM00320         854\nTMhelix         539\nPF00400         450\nPS50082         435\n3.40.50.300     402\nSSF52540        383\nSM00355         285\n2.130.10.10     233\nPS50157         226\nSSF48371        226\n3.30.160.60     220\nPS00028         216\nPR00320         180\nSM00028         168\nPS00678         157\nSSF57667        153\nSSF50978        149\nPS50294         147\n3.30.70.330     145\nPS50005         139\nSSF54928        136\nSM00360         133\nSM00386         132\nPS50102         130\nPF00076         128\n1.25.10.10      122\n1.25.40.10      117\nSSF48452        114\nPF00096         108\n2.40.50.140     102\nSM00382         101\nSSF50249        100\nPS51194          93\nPS51192          93\nSM00490          93\nSM00487          91\nPF00271          89\nPS51450          83\nPR00315          81\ncd00009          71\nSSF46785         69\nPF00270          67\ncd00200          65\nSSF56112         65\n3.30.40.10       64\nPR00304          60\nPR00301          58\nName: id, dtype: int64"
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "translation_overlap_df['id'].value_counts().head(50)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "## Output"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%% md\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "outputs": [],
   "source": [
    "transcript_sequence_df.to_csv('../data/interim/aa_seqs.csv', index=False)\n",
    "translation_overlap_df.to_csv('../data/interim/protein_domains.csv', index=False)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}